<!DOCTYPE html>
<html lang="en">
<head>
  
  <meta charset="utf-8">
  <title>Using gaston for estimating heritability-1</title>
  <meta name="description" content="Using gaston for estimating heritability-1">
  <meta name="author" content="Martha Aquino">
  
  <meta name="viewport" content="width=device-width, initial-scale=1">
  
  <noscript>
    <link href="//fonts.googleapis.com/css?family=Source+Sans+Pro:400,600" rel="stylesheet" type="text/css">
    <link rel="stylesheet" href="/css/fonts.css">
  </noscript>
  
  
  <style type="text/css" media="screen">
    /*!
Pure v1.0.0
Copyright 2013 Yahoo!
Licensed under the BSD License.
https://github.com/yahoo/pure/blob/master/LICENSE.md
*/
/*!
normalize.css v^3.0 | MIT License | git.io/normalize
Copyright (c) Nicolas Gallagher and Jonathan Neal
*/
/*! normalize.css v3.0.3 | MIT License | github.com/necolas/normalize.css */img,legend{border:0}legend,td,th{padding:0}html{font-family:sans-serif;-ms-text-size-adjust:100%;-webkit-text-size-adjust:100%}body{margin:0}article,aside,details,figcaption,figure,footer,header,hgroup,main,menu,nav,section,summary{display:block}audio,canvas,progress,video{display:inline-block;vertical-align:baseline}audio:not([controls]){display:none;height:0}[hidden],template{display:none}a{background-color:transparent}a:active,a:hover{outline:0}abbr[title]{border-bottom:1px dotted}b,optgroup,strong{font-weight:700}dfn{font-style:italic}h1{font-size:2em;margin:.67em 0}mark{background:#ff0;color:#000}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sup{top:-.5em}sub{bottom:-.25em}svg:not(:root){overflow:hidden}figure{margin:1em 40px}hr{box-sizing:content-box;height:0}pre,textarea{overflow:auto}code,kbd,pre,samp{font-family:monospace,monospace;font-size:1em}button,input,optgroup,select,textarea{color:inherit;font:inherit;margin:0}button{overflow:visible}button,select{text-transform:none}button,html input[type=button],input[type=reset],input[type=submit]{-webkit-appearance:button;cursor:pointer}button[disabled],html input[disabled]{cursor:default}button::-moz-focus-inner,input::-moz-focus-inner{border:0;padding:0}input{line-height:normal}input[type=checkbox],input[type=radio]{box-sizing:border-box;padding:0}input[type=number]::-webkit-inner-spin-button,input[type=number]::-webkit-outer-spin-button{height:auto}input[type=search]{-webkit-appearance:textfield;box-sizing:content-box}input[type=search]::-webkit-search-cancel-button,input[type=search]::-webkit-search-decoration{-webkit-appearance:none}fieldset{border:1px solid silver;margin:0 2px;padding:.35em .625em .75em}table{border-collapse:collapse;border-spacing:0}.hidden,[hidden]{display:none!important}.pure-img{max-width:100%;height:auto;display:block}
  </style>
  <style type="text/css" media="screen">
    /*!
Pure v1.0.0
Copyright 2013 Yahoo!
Licensed under the BSD License.
https://github.com/yahoo/pure/blob/master/LICENSE.md
*/
.pure-menu{box-sizing:border-box}.pure-menu-fixed{position:fixed;left:0;top:0;z-index:3}.pure-menu-item,.pure-menu-list{position:relative}.pure-menu-list{list-style:none;margin:0;padding:0}.pure-menu-item{padding:0;margin:0;height:100%}.pure-menu-heading,.pure-menu-link{display:block;text-decoration:none;white-space:nowrap}.pure-menu-horizontal{width:100%;white-space:normal}.pure-menu-horizontal .pure-menu-list{display:inline-block}.pure-menu-horizontal .pure-menu-heading,.pure-menu-horizontal .pure-menu-item,.pure-menu-horizontal .pure-menu-separator{display:inline-block;zoom:1;vertical-align:middle}.pure-menu-item .pure-menu-item{display:block}.pure-menu-children{display:none;position:absolute;left:100%;top:0;margin:0;padding:0;z-index:3}.pure-menu-horizontal .pure-menu-children{left:0;top:auto;width:inherit}.pure-menu-active>.pure-menu-children,.pure-menu-allow-hover:hover>.pure-menu-children{display:block;position:absolute}.pure-menu-has-children>.pure-menu-link:after{padding-left:.5em;content:"\25B8";font-size:small}.pure-menu-horizontal .pure-menu-has-children>.pure-menu-link:after{content:"\25BE"}.pure-menu-scrollable{overflow-y:scroll;overflow-x:hidden}.pure-menu-scrollable .pure-menu-list{display:block}.pure-menu-horizontal.pure-menu-scrollable .pure-menu-list{display:inline-block}.pure-menu-horizontal.pure-menu-scrollable{white-space:nowrap;overflow-y:hidden;overflow-x:auto;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;padding:.5em 0}.pure-menu-horizontal.pure-menu-scrollable::-webkit-scrollbar{display:none}.pure-menu-horizontal .pure-menu-children .pure-menu-separator,.pure-menu-separator{background-color:#ccc;height:1px;margin:.3em 0}.pure-menu-horizontal .pure-menu-separator{width:1px;height:1.3em;margin:0 .3em}.pure-menu-horizontal .pure-menu-children .pure-menu-separator{display:block;width:auto}.pure-menu-heading{text-transform:uppercase;color:#565d64}.pure-menu-link{color:#777}.pure-menu-children{background-color:#fff}.pure-menu-disabled,.pure-menu-heading,.pure-menu-link{padding:.5em 1em}.pure-menu-disabled{opacity:.5}.pure-menu-disabled .pure-menu-link:hover{background-color:transparent}.pure-menu-active>.pure-menu-link,.pure-menu-link:focus,.pure-menu-link:hover{background-color:#eee}.pure-menu-selected .pure-menu-link,.pure-menu-selected .pure-menu-link:visited{color:#000}

  </style>
  <style type="text/css" media="screen">
    /*!
Pure v1.0.0
Copyright 2013 Yahoo!
Licensed under the BSD License.
https://github.com/yahoo/pure/blob/master/LICENSE.md
*/
.pure-g{letter-spacing:-.31em;text-rendering:optimizespeed;font-family:FreeSans,Arimo,"Droid Sans",Helvetica,Arial,sans-serif;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-flow:row wrap;-ms-flex-flow:row wrap;flex-flow:row wrap;-webkit-align-content:flex-start;-ms-flex-line-pack:start;align-content:flex-start}@media all and (-ms-high-contrast:none),(-ms-high-contrast:active){table .pure-g{display:block}}.opera-only :-o-prefocus,.pure-g{word-spacing:-.43em}.pure-u,.pure-u-1,.pure-u-1-1,.pure-u-1-12,.pure-u-1-2,.pure-u-1-24,.pure-u-1-3,.pure-u-1-4,.pure-u-1-5,.pure-u-1-6,.pure-u-1-8,.pure-u-10-24,.pure-u-11-12,.pure-u-11-24,.pure-u-12-24,.pure-u-13-24,.pure-u-14-24,.pure-u-15-24,.pure-u-16-24,.pure-u-17-24,.pure-u-18-24,.pure-u-19-24,.pure-u-2-24,.pure-u-2-3,.pure-u-2-5,.pure-u-20-24,.pure-u-21-24,.pure-u-22-24,.pure-u-23-24,.pure-u-24-24,.pure-u-3-24,.pure-u-3-4,.pure-u-3-5,.pure-u-3-8,.pure-u-4-24,.pure-u-4-5,.pure-u-5-12,.pure-u-5-24,.pure-u-5-5,.pure-u-5-6,.pure-u-5-8,.pure-u-6-24,.pure-u-7-12,.pure-u-7-24,.pure-u-7-8,.pure-u-8-24,.pure-u-9-24{display:inline-block;zoom:1;letter-spacing:normal;word-spacing:normal;vertical-align:top;text-rendering:auto}.pure-g [class*=pure-u]{font-family:sans-serif}.pure-u-1-24{width:4.1667%}.pure-u-1-12,.pure-u-2-24{width:8.3333%}.pure-u-1-8,.pure-u-3-24{width:12.5%}.pure-u-1-6,.pure-u-4-24{width:16.6667%}.pure-u-1-5{width:20%}.pure-u-5-24{width:20.8333%}.pure-u-1-4,.pure-u-6-24{width:25%}.pure-u-7-24{width:29.1667%}.pure-u-1-3,.pure-u-8-24{width:33.3333%}.pure-u-3-8,.pure-u-9-24{width:37.5%}.pure-u-2-5{width:40%}.pure-u-10-24,.pure-u-5-12{width:41.6667%}.pure-u-11-24{width:45.8333%}.pure-u-1-2,.pure-u-12-24{width:50%}.pure-u-13-24{width:54.1667%}.pure-u-14-24,.pure-u-7-12{width:58.3333%}.pure-u-3-5{width:60%}.pure-u-15-24,.pure-u-5-8{width:62.5%}.pure-u-16-24,.pure-u-2-3{width:66.6667%}.pure-u-17-24{width:70.8333%}.pure-u-18-24,.pure-u-3-4{width:75%}.pure-u-19-24{width:79.1667%}.pure-u-4-5{width:80%}.pure-u-20-24,.pure-u-5-6{width:83.3333%}.pure-u-21-24,.pure-u-7-8{width:87.5%}.pure-u-11-12,.pure-u-22-24{width:91.6667%}.pure-u-23-24{width:95.8333%}.pure-u-1,.pure-u-1-1,.pure-u-24-24,.pure-u-5-5{width:100%}
  </style>
  
  
  <style type="text/css" media="screen">
    /*!
Pure v1.0.0
Copyright 2013 Yahoo!
Licensed under the BSD License.
https://github.com/yahoo/pure/blob/master/LICENSE.md
*/
@media screen and (min-width:35.5em){.pure-u-sm-1,.pure-u-sm-1-1,.pure-u-sm-1-12,.pure-u-sm-1-2,.pure-u-sm-1-24,.pure-u-sm-1-3,.pure-u-sm-1-4,.pure-u-sm-1-5,.pure-u-sm-1-6,.pure-u-sm-1-8,.pure-u-sm-10-24,.pure-u-sm-11-12,.pure-u-sm-11-24,.pure-u-sm-12-24,.pure-u-sm-13-24,.pure-u-sm-14-24,.pure-u-sm-15-24,.pure-u-sm-16-24,.pure-u-sm-17-24,.pure-u-sm-18-24,.pure-u-sm-19-24,.pure-u-sm-2-24,.pure-u-sm-2-3,.pure-u-sm-2-5,.pure-u-sm-20-24,.pure-u-sm-21-24,.pure-u-sm-22-24,.pure-u-sm-23-24,.pure-u-sm-24-24,.pure-u-sm-3-24,.pure-u-sm-3-4,.pure-u-sm-3-5,.pure-u-sm-3-8,.pure-u-sm-4-24,.pure-u-sm-4-5,.pure-u-sm-5-12,.pure-u-sm-5-24,.pure-u-sm-5-5,.pure-u-sm-5-6,.pure-u-sm-5-8,.pure-u-sm-6-24,.pure-u-sm-7-12,.pure-u-sm-7-24,.pure-u-sm-7-8,.pure-u-sm-8-24,.pure-u-sm-9-24{display:inline-block;zoom:1;letter-spacing:normal;word-spacing:normal;vertical-align:top;text-rendering:auto}.pure-u-sm-1-24{width:4.1667%}.pure-u-sm-1-12,.pure-u-sm-2-24{width:8.3333%}.pure-u-sm-1-8,.pure-u-sm-3-24{width:12.5%}.pure-u-sm-1-6,.pure-u-sm-4-24{width:16.6667%}.pure-u-sm-1-5{width:20%}.pure-u-sm-5-24{width:20.8333%}.pure-u-sm-1-4,.pure-u-sm-6-24{width:25%}.pure-u-sm-7-24{width:29.1667%}.pure-u-sm-1-3,.pure-u-sm-8-24{width:33.3333%}.pure-u-sm-3-8,.pure-u-sm-9-24{width:37.5%}.pure-u-sm-2-5{width:40%}.pure-u-sm-10-24,.pure-u-sm-5-12{width:41.6667%}.pure-u-sm-11-24{width:45.8333%}.pure-u-sm-1-2,.pure-u-sm-12-24{width:50%}.pure-u-sm-13-24{width:54.1667%}.pure-u-sm-14-24,.pure-u-sm-7-12{width:58.3333%}.pure-u-sm-3-5{width:60%}.pure-u-sm-15-24,.pure-u-sm-5-8{width:62.5%}.pure-u-sm-16-24,.pure-u-sm-2-3{width:66.6667%}.pure-u-sm-17-24{width:70.8333%}.pure-u-sm-18-24,.pure-u-sm-3-4{width:75%}.pure-u-sm-19-24{width:79.1667%}.pure-u-sm-4-5{width:80%}.pure-u-sm-20-24,.pure-u-sm-5-6{width:83.3333%}.pure-u-sm-21-24,.pure-u-sm-7-8{width:87.5%}.pure-u-sm-11-12,.pure-u-sm-22-24{width:91.6667%}.pure-u-sm-23-24{width:95.8333%}.pure-u-sm-1,.pure-u-sm-1-1,.pure-u-sm-24-24,.pure-u-sm-5-5{width:100%}}@media screen and (min-width:48em){.pure-u-md-1,.pure-u-md-1-1,.pure-u-md-1-12,.pure-u-md-1-2,.pure-u-md-1-24,.pure-u-md-1-3,.pure-u-md-1-4,.pure-u-md-1-5,.pure-u-md-1-6,.pure-u-md-1-8,.pure-u-md-10-24,.pure-u-md-11-12,.pure-u-md-11-24,.pure-u-md-12-24,.pure-u-md-13-24,.pure-u-md-14-24,.pure-u-md-15-24,.pure-u-md-16-24,.pure-u-md-17-24,.pure-u-md-18-24,.pure-u-md-19-24,.pure-u-md-2-24,.pure-u-md-2-3,.pure-u-md-2-5,.pure-u-md-20-24,.pure-u-md-21-24,.pure-u-md-22-24,.pure-u-md-23-24,.pure-u-md-24-24,.pure-u-md-3-24,.pure-u-md-3-4,.pure-u-md-3-5,.pure-u-md-3-8,.pure-u-md-4-24,.pure-u-md-4-5,.pure-u-md-5-12,.pure-u-md-5-24,.pure-u-md-5-5,.pure-u-md-5-6,.pure-u-md-5-8,.pure-u-md-6-24,.pure-u-md-7-12,.pure-u-md-7-24,.pure-u-md-7-8,.pure-u-md-8-24,.pure-u-md-9-24{display:inline-block;zoom:1;letter-spacing:normal;word-spacing:normal;vertical-align:top;text-rendering:auto}.pure-u-md-1-24{width:4.1667%}.pure-u-md-1-12,.pure-u-md-2-24{width:8.3333%}.pure-u-md-1-8,.pure-u-md-3-24{width:12.5%}.pure-u-md-1-6,.pure-u-md-4-24{width:16.6667%}.pure-u-md-1-5{width:20%}.pure-u-md-5-24{width:20.8333%}.pure-u-md-1-4,.pure-u-md-6-24{width:25%}.pure-u-md-7-24{width:29.1667%}.pure-u-md-1-3,.pure-u-md-8-24{width:33.3333%}.pure-u-md-3-8,.pure-u-md-9-24{width:37.5%}.pure-u-md-2-5{width:40%}.pure-u-md-10-24,.pure-u-md-5-12{width:41.6667%}.pure-u-md-11-24{width:45.8333%}.pure-u-md-1-2,.pure-u-md-12-24{width:50%}.pure-u-md-13-24{width:54.1667%}.pure-u-md-14-24,.pure-u-md-7-12{width:58.3333%}.pure-u-md-3-5{width:60%}.pure-u-md-15-24,.pure-u-md-5-8{width:62.5%}.pure-u-md-16-24,.pure-u-md-2-3{width:66.6667%}.pure-u-md-17-24{width:70.8333%}.pure-u-md-18-24,.pure-u-md-3-4{width:75%}.pure-u-md-19-24{width:79.1667%}.pure-u-md-4-5{width:80%}.pure-u-md-20-24,.pure-u-md-5-6{width:83.3333%}.pure-u-md-21-24,.pure-u-md-7-8{width:87.5%}.pure-u-md-11-12,.pure-u-md-22-24{width:91.6667%}.pure-u-md-23-24{width:95.8333%}.pure-u-md-1,.pure-u-md-1-1,.pure-u-md-24-24,.pure-u-md-5-5{width:100%}}@media screen and (min-width:64em){.pure-u-lg-1,.pure-u-lg-1-1,.pure-u-lg-1-12,.pure-u-lg-1-2,.pure-u-lg-1-24,.pure-u-lg-1-3,.pure-u-lg-1-4,.pure-u-lg-1-5,.pure-u-lg-1-6,.pure-u-lg-1-8,.pure-u-lg-10-24,.pure-u-lg-11-12,.pure-u-lg-11-24,.pure-u-lg-12-24,.pure-u-lg-13-24,.pure-u-lg-14-24,.pure-u-lg-15-24,.pure-u-lg-16-24,.pure-u-lg-17-24,.pure-u-lg-18-24,.pure-u-lg-19-24,.pure-u-lg-2-24,.pure-u-lg-2-3,.pure-u-lg-2-5,.pure-u-lg-20-24,.pure-u-lg-21-24,.pure-u-lg-22-24,.pure-u-lg-23-24,.pure-u-lg-24-24,.pure-u-lg-3-24,.pure-u-lg-3-4,.pure-u-lg-3-5,.pure-u-lg-3-8,.pure-u-lg-4-24,.pure-u-lg-4-5,.pure-u-lg-5-12,.pure-u-lg-5-24,.pure-u-lg-5-5,.pure-u-lg-5-6,.pure-u-lg-5-8,.pure-u-lg-6-24,.pure-u-lg-7-12,.pure-u-lg-7-24,.pure-u-lg-7-8,.pure-u-lg-8-24,.pure-u-lg-9-24{display:inline-block;zoom:1;letter-spacing:normal;word-spacing:normal;vertical-align:top;text-rendering:auto}.pure-u-lg-1-24{width:4.1667%}.pure-u-lg-1-12,.pure-u-lg-2-24{width:8.3333%}.pure-u-lg-1-8,.pure-u-lg-3-24{width:12.5%}.pure-u-lg-1-6,.pure-u-lg-4-24{width:16.6667%}.pure-u-lg-1-5{width:20%}.pure-u-lg-5-24{width:20.8333%}.pure-u-lg-1-4,.pure-u-lg-6-24{width:25%}.pure-u-lg-7-24{width:29.1667%}.pure-u-lg-1-3,.pure-u-lg-8-24{width:33.3333%}.pure-u-lg-3-8,.pure-u-lg-9-24{width:37.5%}.pure-u-lg-2-5{width:40%}.pure-u-lg-10-24,.pure-u-lg-5-12{width:41.6667%}.pure-u-lg-11-24{width:45.8333%}.pure-u-lg-1-2,.pure-u-lg-12-24{width:50%}.pure-u-lg-13-24{width:54.1667%}.pure-u-lg-14-24,.pure-u-lg-7-12{width:58.3333%}.pure-u-lg-3-5{width:60%}.pure-u-lg-15-24,.pure-u-lg-5-8{width:62.5%}.pure-u-lg-16-24,.pure-u-lg-2-3{width:66.6667%}.pure-u-lg-17-24{width:70.8333%}.pure-u-lg-18-24,.pure-u-lg-3-4{width:75%}.pure-u-lg-19-24{width:79.1667%}.pure-u-lg-4-5{width:80%}.pure-u-lg-20-24,.pure-u-lg-5-6{width:83.3333%}.pure-u-lg-21-24,.pure-u-lg-7-8{width:87.5%}.pure-u-lg-11-12,.pure-u-lg-22-24{width:91.6667%}.pure-u-lg-23-24{width:95.8333%}.pure-u-lg-1,.pure-u-lg-1-1,.pure-u-lg-24-24,.pure-u-lg-5-5{width:100%}}@media screen and (min-width:80em){.pure-u-xl-1,.pure-u-xl-1-1,.pure-u-xl-1-12,.pure-u-xl-1-2,.pure-u-xl-1-24,.pure-u-xl-1-3,.pure-u-xl-1-4,.pure-u-xl-1-5,.pure-u-xl-1-6,.pure-u-xl-1-8,.pure-u-xl-10-24,.pure-u-xl-11-12,.pure-u-xl-11-24,.pure-u-xl-12-24,.pure-u-xl-13-24,.pure-u-xl-14-24,.pure-u-xl-15-24,.pure-u-xl-16-24,.pure-u-xl-17-24,.pure-u-xl-18-24,.pure-u-xl-19-24,.pure-u-xl-2-24,.pure-u-xl-2-3,.pure-u-xl-2-5,.pure-u-xl-20-24,.pure-u-xl-21-24,.pure-u-xl-22-24,.pure-u-xl-23-24,.pure-u-xl-24-24,.pure-u-xl-3-24,.pure-u-xl-3-4,.pure-u-xl-3-5,.pure-u-xl-3-8,.pure-u-xl-4-24,.pure-u-xl-4-5,.pure-u-xl-5-12,.pure-u-xl-5-24,.pure-u-xl-5-5,.pure-u-xl-5-6,.pure-u-xl-5-8,.pure-u-xl-6-24,.pure-u-xl-7-12,.pure-u-xl-7-24,.pure-u-xl-7-8,.pure-u-xl-8-24,.pure-u-xl-9-24{display:inline-block;zoom:1;letter-spacing:normal;word-spacing:normal;vertical-align:top;text-rendering:auto}.pure-u-xl-1-24{width:4.1667%}.pure-u-xl-1-12,.pure-u-xl-2-24{width:8.3333%}.pure-u-xl-1-8,.pure-u-xl-3-24{width:12.5%}.pure-u-xl-1-6,.pure-u-xl-4-24{width:16.6667%}.pure-u-xl-1-5{width:20%}.pure-u-xl-5-24{width:20.8333%}.pure-u-xl-1-4,.pure-u-xl-6-24{width:25%}.pure-u-xl-7-24{width:29.1667%}.pure-u-xl-1-3,.pure-u-xl-8-24{width:33.3333%}.pure-u-xl-3-8,.pure-u-xl-9-24{width:37.5%}.pure-u-xl-2-5{width:40%}.pure-u-xl-10-24,.pure-u-xl-5-12{width:41.6667%}.pure-u-xl-11-24{width:45.8333%}.pure-u-xl-1-2,.pure-u-xl-12-24{width:50%}.pure-u-xl-13-24{width:54.1667%}.pure-u-xl-14-24,.pure-u-xl-7-12{width:58.3333%}.pure-u-xl-3-5{width:60%}.pure-u-xl-15-24,.pure-u-xl-5-8{width:62.5%}.pure-u-xl-16-24,.pure-u-xl-2-3{width:66.6667%}.pure-u-xl-17-24{width:70.8333%}.pure-u-xl-18-24,.pure-u-xl-3-4{width:75%}.pure-u-xl-19-24{width:79.1667%}.pure-u-xl-4-5{width:80%}.pure-u-xl-20-24,.pure-u-xl-5-6{width:83.3333%}.pure-u-xl-21-24,.pure-u-xl-7-8{width:87.5%}.pure-u-xl-11-12,.pure-u-xl-22-24{width:91.6667%}.pure-u-xl-23-24{width:95.8333%}.pure-u-xl-1,.pure-u-xl-1-1,.pure-u-xl-24-24,.pure-u-xl-5-5{width:100%}}
  </style>
  

  
  <style type="text/css" media="screen">
    html {
	min-height:100%;
	width:100%;
	position:relative;
}

body {
	background-color: rgb(252,252,252);
	color: #484848;
}

.nav-menu {
  margin-top: 5px;
  padding-bottom: 5px;
	border-bottom: 1px solid #e3e3e3;
}

.pure-menu-heading {
  text-transform: none;
  font-size: large;
}

.header {
	text-align: left;
	color: #484848;
	margin-bottom: .5em;
}

.header ul li {
		height: auto;
}

.header ul li a {
	font-weight: bold;
	color: #484848;
	font-family: "Source Sans Pro", "HelveticaNeue", "Helvetica Neue", Helvetica, Arial, sans-serif;
}

.header {
	font-weight: bold;
	color: #484848;
	font-family: "Source Sans Pro", "HelveticaNeue", "Helvetica Neue", Helvetica, Arial, sans-serif;
}

.site-title {
	color: #484848;
  text-transform: none;
  font-weight: normal;
	font-family: "Source Sans Pro", "HelveticaNeue", "Helvetica Neue", Helvetica, Arial, sans-serif;
}

.pull-right {
	float: right;
}

.posts-name {
	text-transform: capitalize;
	font-weight: bold;
	padding-left: 1em;
	margin-top: 1em;
	font-family: "Source Sans Pro", "HelveticaNeue", "Helvetica Neue", Helvetica, Arial, sans-serif;
}

.posts {
	font-family: verdana,arial,helvetica,sans-serif;
	list-style-type:none;
	padding-left: 1em;
}

.posts li p {
	margin-top: 0;
}

.posts li {
	margin-bottom: 1em;
}

.posts li a {
	color:#369;
	text-decoration:none;
}

.post-list {
  font-size: large;
}

.footnote {
	font-family: verdana,arial,helvetica,sans-serif;
	color:#888;
	font-size:x-small;
	margin-bottom:0;
}

.footnote a {
	color: #888 !important
}

.footnote a:hover{
	text-decoration:underline;
	color:#369;
}

.footer {
	position:absolute;
	z-index:2;
	height:auto;
	width:100%;
	bottom:0;
}

.footer-content {
	border-top: 1px solid #e3e3e3;
	font-size: 80%;
	color: #bbb;
}

.footer-content a {
	color: #bbb;
}

.footer-content ul {
	height: auto;
	margin-top: 0;
	margin-bottom: 0;
	display: inline-block;
	padding-left: 0;
}

#gototop-btn {
	display: inline-block;
}

#foot-name {
  color: #484848;
	text-transform: none;
}

#foot-copyright {
	padding-left: 1em;
	padding-bottom: 0.5em;
	margin: 0;
}

.post {
	font-family: proxima-nova,"Helvetica Neue",Helvetica,Roboto,Arial,sans-serif;
	color:#484848;
	letter-spacing:normal;
	padding-left: .5em;
}

.post h1,h2,h3,h4,h5,h6 {
	font-weight: bold;
	letter-spacing:normal;
}

.post-content {
	z-index:9;
	overflow:auto;
	padding:0;
	padding-bottom:3em;
	font-size:16px;
	line-height:1.4;
}

.post-content img {
	max-width: 100%;
	height: auto;
}

/* these sections screw up other themes */ 
/*.post-content code {
  background-color: #EEE;
}

.post code {
  background-color: #EEE;
} */
/* to here */

.post a {
	color:#c05b4d;
	text-decoration: none;
}

.post a:hover {
	color:#a5473a;
	text-decoration: underline;
}

.post h1 {
	font-size:28px;
}

.post h2 {
	font-size:25px;
}

.post h3 {
	font-size:23px;
}

.post h4 {
	font-size:21px;
}

.post h5 {
	font-size:19px;
}

.post h6 {
	font-size:18px;
}

.post-title{
		margin-top: 0;
		margin-bottom: 2em;
}

.post-title h1 {
	font-weight:bold;
	font-size:39px;
	line-height: 40px;
	margin-top: 0;
	margin-bottom: 0;
}

@media screen and (max-width:767px){
	.desktop {
		display: none;
	}
	.mobile {
		display: block;
	}
	#toggle-btn {
		display: inline-block;
		float: right;
		padding: .5em 1em;
		text-decoration: none;
		color: #484848;
		font-weight: bold;
	}
	#toggle-content li {
		clear: both;
		height: auto;
		background-color: rgb(249,249,249);
	}
	#toggle-home {
		display: inline-block;
	}
}

@media screen and (min-width:768px){
	.mobile {
		display:none;
	}
	.desktop {
		display: block;
	}
}

  </style>
  
  <style type="text/css" media="screen"> 
    
      /*

Zenburn style from voldmar.ru (c) Vladimir Epifanov <voldmar@voldmar.ru>
based on dark.css by Ivan Sagalaev

*/

.hljs {
  display: block;
  overflow-x: auto;
  padding: 0.5em;
  background: #3f3f3f;
  color: #dcdcdc;
}

.hljs-keyword,
.hljs-selector-tag,
.hljs-tag {
  color: #e3ceab;
}

.hljs-template-tag {
  color: #dcdcdc;
}

.hljs-number {
  color: #8cd0d3;
}

.hljs-variable,
.hljs-template-variable,
.hljs-attribute {
  color: #efdcbc;
}

.hljs-literal {
  color: #efefaf;
}

.hljs-subst {
  color: #8f8f8f;
}

.hljs-title,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-section,
.hljs-type {
  color: #efef8f;
}

.hljs-symbol,
.hljs-bullet,
.hljs-link {
  color: #dca3a3;
}

.hljs-deletion,
.hljs-string,
.hljs-built_in,
.hljs-builtin-name {
  color: #cc9393;
}

.hljs-addition,
.hljs-comment,
.hljs-quote,
.hljs-meta {
  color: #7f9f7f;
}


.hljs-emphasis {
  font-style: italic;
}

.hljs-strong {
  font-weight: bold;
}
 
    
  </style>
  

  
  <link rel="shortcut icon" 
   
    href="/img/favicon.ico"
  
  type="image/x-icon">

  
  

  <meta name="generator" content="Hugo 0.40.3" />

  
  
</head>
<body>

<div class="header pure-g">
    <div class="pure-u-1-24 pure-u-md-1-6"></div>
    <div class="pure-u-11-12 pure-u-md-2-3">
        <div class="desktop pure-menu pure-menu-horizontal nav-menu">
            
            <a href="/" class="site-title pure-menu-heading">Electronic Lab Notebook</a>
            <ul class="pure-menu-list pull-right">
				
                
                <li class="pure-menu-item">
                    <a href="/categories/coding" class="pure-menu-link">Coding</a>
                </li>
                
                <li class="pure-menu-item">
                    <a href="/categories/experiments" class="pure-menu-link">Experiments</a>
                </li>
                
                <li class="pure-menu-item">
                    <a href="/categories/reading" class="pure-menu-link">Reading</a>
                </li>
                
				
                <li class="pure-menu-item">
                    <a href="/about/" class="pure-menu-link">About</a>
                </li>
            </ul>
        </div>
        <div class="mobile pure-menu nav-menu">
            <a href="/" class="site-title pure-menu-heading" id="toggle-home">Electronic Lab Notebook</a>
            <a href="#" id="toggle-btn">&#9776;</a>
            <ul class="pure-menu-list" id="toggle-content" style="display:none;">
                
				
                
                <li class="pure-menu-item">
                    <a href="/categories/coding" class="pure-menu-link">Coding</a>
                </li>
                
                <li class="pure-menu-item">
                    <a href="/categories/experiments" class="pure-menu-link">Experiments</a>
                </li>
                
                <li class="pure-menu-item">
                    <a href="/categories/reading" class="pure-menu-link">Reading</a>
                </li>
                
				
                <li class="pure-menu-item">
                    <a href="/about" class="pure-menu-link">About</a>
                </li>
            </ul>
        </div>
    </div>
    <div class="pure-u-1-24 pure-u-md-1-6"></div>
</div>


<div class="pure-g">
  <div class="pure-u-1-24 pure-u-md-1-6"></div>
  <div class="pure-u-11-12 pure-u-md-2-3">
    <div class="post">

      <div class="post-title">
        <p class="footnote">
        <time class="">08-19-2018</time>
        
        
        |
        
        
        tags: [ <a href="/tags/gaston">Gaston</a> <a href="/tags/genomic-data">Genomic Data</a> <a href="/tags/r">R</a> <a href="/tags/plink">PLINK</a> <a href="/tags/heritability">Heritability</a> ]
        

        
        categories: [<a href="/categories/coding">Coding</a> <a href="/categories/experiments">Experiments</a> ]
        

        
        </p>
        <h1>Using gaston for estimating heritability-1</h1>
      </div>

      <div class="post-content">
        <div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The genetic relationship matrix generated from the previous exercise contains values that signify the degree of relatedness between the individuals. This will allow me to estimate the heritability of the traits and principal components identified in the first objective.</p>
<p>To recap this concept, traits can vary among individuals due to environmental and genetic factors. Estimating the heritability of a trait means determining the proportion of variance that can be attributed to genetics (<a href="https://www.nature.com/scitable/topicpage/estimating-trait-heritability-46889" class="uri">https://www.nature.com/scitable/topicpage/estimating-trait-heritability-46889</a>). The relationship matrix is used to account for the degree of relatedness between the individuals when considering phenotypic variance. This is how heritabiity estimates are calculated.</p>
</div>
<div id="methods-and-results" class="section level1">
<h1>Methods and Results</h1>
<div id="load-imputed-phenotypic-data" class="section level3">
<h3>1. Load imputed phenotypic data</h3>
<pre class="r"><code>imputed_phen_data &lt;- read.csv(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/imputed_phenotypic_data_uvaf.csv&#39;, header = T)

head(imputed_phen_data)</code></pre>
<pre><code>##     UUID R.Logmar.VA L.Logmar.VA R.Sph..pre.dilate. R.Cyl..pre.dilate.
## 1 219960        0.02       -0.04               0.25               0.00
## 2 313180        0.10        0.16               0.00              -0.75
## 3 314911        0.00        0.00               1.25              -1.25
## 4 111150        0.30        0.08               1.25              -0.25
## 5 363161        0.00       -0.10               1.25              -0.25
## 6 110460        0.24        0.36               4.00              -0.50
##   R.Axis..pre.dilate. L.Sph..pre.dilate. L.Cyl..pre.dilate.
## 1                   0               0.25              -0.50
## 2                  28              -0.50              -0.25
## 3                 148               1.25              -0.25
## 4                  97               1.50              -0.50
## 5                  37               1.25              -0.75
## 6                  46               3.75              -0.75
##   L.Axis..pre.dilate. R.K.value.H R.K.Value.H.Axis R.K.value.V
## 1                  79    42.00000           2.0000    43.00000
## 2                 164    41.25000           7.0000    42.25000
## 3                  24    43.91437         145.9032    44.91517
## 4                  81    44.75000           5.0000    45.00000
## 5                 164    44.75000           0.0000    44.75000
## 6                 180    42.00000           8.0000    43.25000
##   R.K.value.V.Axis L.K.value.H L.K.value.H.Axis L.K.value.V
## 1         92.00000    42.50000          5.00000    43.50000
## 2         97.00000    41.50000        168.00000    42.00000
## 3         49.64518    44.06458         38.39706    44.88776
## 4         95.00000    45.00000         60.00000    45.25000
## 5         90.00000    44.25000        178.00000    44.75000
## 6         98.00000    42.25000        177.00000    43.25000
##   L.K.value.V.Axis R.Pachimetry L.Pachimetry R.Axial.Length L.Axial.Length
## 1          95.0000          532          554          24.31          24.10
## 2          78.0000          608          612          25.02          25.21
## 3         137.2832          507          510          22.78          22.80
## 4         150.0000          560          559          23.02          22.98
## 5          88.0000          556          562          21.75          22.04
## 6          87.0000          498          501          23.06          23.17
##   R.AC.Depth L.AC.Depth R.IOP.mmHg L.IOP.mmHg R.CDR L.CDR totaluvafmm
## 1       3.09       3.03         14         14   0.9   0.9     0.00000
## 2       3.38       3.92         16         15   0.9   0.7    74.40000
## 3       3.40       3.45         26         22   0.7   0.7    14.60000
## 4       3.00       2.85         14         14   0.2   0.2     1.90000
## 5       2.60       2.53         22         21   0.3   0.3    23.98903
## 6       2.94       3.04         18         20   0.6   0.6     1.00000</code></pre>
</div>
<div id="load-genomic-data" class="section level3">
<h3>2. Load genomic data</h3>
<pre class="r"><code>merged_nies_fam &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/merged_nies_filtered.fam&#39;, header = F)

head(merged_nies_fam)</code></pre>
<pre><code>##   V1     V2 V3 V4 V5 V6
## 1  1 110150  0  0  2 -9
## 2  1 110160  0  0  1 -9
## 3  1 110500  0  0  2 -9
## 4  1 110650  0  0  2 -9
## 5  1 110820  0  0  2 -9
## 6  1 111280  0  0  2 -9</code></pre>
</div>
<div id="extract-uuids-in-genomic-data" class="section level3">
<h3>3. Extract UUIDs in genomic data</h3>
<pre class="r"><code>geno_data_uuid &lt;- merged_nies_fam$V2</code></pre>
</div>
<div id="extract-relevant-phenotypic-data-based-on-genomic-uuids" class="section level3">
<h3>4. Extract relevant phenotypic data based on genomic UUIDs</h3>
<pre class="r"><code>nies_pheno &lt;- imputed_phen_data[imputed_phen_data$UUID %in% c(geno_data_uuid), ]
head(nies_pheno)</code></pre>
<pre><code>##      UUID R.Logmar.VA L.Logmar.VA R.Sph..pre.dilate. R.Cyl..pre.dilate.
## 1  219960        0.02       -0.04               0.25               0.00
## 2  313180        0.10        0.16               0.00              -0.75
## 7  320511       -0.06        0.02               1.00               0.00
## 11 400011       -0.14       -0.10               0.75              -0.25
## 13 400013        0.30        0.30               1.50              -0.25
## 14 316131        0.08        0.26               2.00              -1.00
##    R.Axis..pre.dilate. L.Sph..pre.dilate. L.Cyl..pre.dilate.
## 1                    0               0.25              -0.50
## 2                   28              -0.50              -0.25
## 7                    0               1.25              -0.25
## 11                 157               1.25              -0.50
## 13                  82               1.75              -0.50
## 14                  69               1.00              -1.00
##    L.Axis..pre.dilate. R.K.value.H R.K.Value.H.Axis R.K.value.V
## 1                   79       42.00                2       43.00
## 2                  164       41.25                7       42.25
## 7                   29       42.50              177       43.50
## 11                  20       42.50              165       43.25
## 13                 100       43.00                0       43.00
## 14                 114       42.75               63       44.00
##    R.K.value.V.Axis L.K.value.H L.K.value.H.Axis L.K.value.V
## 1                92       42.50                5       43.50
## 2                97       41.50              168       42.00
## 7                87       42.00              170       42.50
## 11               75       42.75               17       43.25
## 13               90       42.75              111       43.00
## 14              153       43.00              115       43.50
##    L.K.value.V.Axis R.Pachimetry L.Pachimetry R.Axial.Length
## 1                95          532          554          24.31
## 2                78          608          612          25.02
## 7                80          499          512          23.37
## 11              107          602          601          23.32
## 13               21          552          550          22.93
## 14               25          583          588          23.49
##    L.Axial.Length R.AC.Depth L.AC.Depth R.IOP.mmHg L.IOP.mmHg R.CDR L.CDR
## 1           24.10       3.09       3.03         14         14   0.9   0.9
## 2           25.21       3.38       3.92         16         15   0.9   0.7
## 7           23.65       3.29       3.08         17         18   0.4   0.4
## 11          23.34       3.81       3.83         15         17   0.6   0.6
## 13          22.70       3.22       3.13         18         18   0.4   0.4
## 14          23.54       3.24       3.04         12         12   0.3   0.3
##    totaluvafmm
## 1          0.0
## 2         74.4
## 7         27.5
## 11        47.0
## 13        20.1
## 14        12.9</code></pre>
<p>I thought that all 359 individuals would have phenotypic data. However, only 346 individuals were extracted, indicating that there are 346 individuals with phenotypic and genomic data. This is most likely because I removed individuals that were &lt;17 years old.</p>
</div>
<div id="isolate-uuids-of-phenotypic-data" class="section level3">
<h3>5. Isolate UUIDs of phenotypic data</h3>
<pre class="r"><code>pheno_data_uuid &lt;- nies_pheno$UUID</code></pre>
<pre class="r"><code>write.table(pheno_data_uuid, &#39;C:/Users/Martha/Documents/Honours/honours.project/Data/filtered_pheno_uuid.txt&#39;, row.names = F, col.names = F, quote = F)</code></pre>
</div>
<div id="extract-genomic-data-for-individuals-with-phenotypic-data" class="section level3">
<h3>6. Extract genomic data for individuals with phenotypic data</h3>
<pre class="bash"><code>plink1.9 --bfile merged_nies_filtered --keep filtered_pheno_uuid2.txt --make-bed --out merged_nies_geno</code></pre>
<p>Note: I added a family ID column to the text file as PLINK only recognises ID files when they have family and individual IDs.</p>
<p>Output:</p>
<ul>
<li><p>346 individuals remain</p></li>
<li><p>13 individuals removed</p></li>
<li><p>3,910,077 variants remain</p></li>
</ul>
</div>
<div id="re-order-samples-in-genomic-data-set-based-on-order-of-uuids-of-phenotypic-data-set" class="section level3">
<h3>7. Re-order samples in genomic data set based on order of UUIDs of phenotypic data set</h3>
<pre class="bash"><code>plink1.9 --bfile merged_nies_geno --indiv-sort f filtered_pheno_uuid2.txt --make-bed --out merged_nies_geno</code></pre>
<p>The samples have to be re-ordered to be the same as the phenotypic data for estimating heritability.</p>
</div>
<div id="run-pca-on-this-new-filtered-data-set" class="section level3">
<h3>8. Run PCA on this new filtered data set</h3>
<pre class="bash"><code>plink1.9 --bfile merged_nies_geno --pca 120 var-wts --out plink_output/nies_geno_pca</code></pre>
<p>PCA and generating GRM have to be repeated for this new data set as it contains less individuals compared to the last data set.</p>
</div>
<div id="load-eigenvalue-file-and-generate-screeplot." class="section level3">
<h3>9. Load eigenvalue file and generate screeplot.</h3>
<pre class="r"><code>nies_geno_eigenval &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/nies_geno_pca.eigenval&#39;, header = F)

barplot(nies_geno_eigenval$V1, 
        names.arg = 1:nrow(nies_geno_eigenval), 
        main = &quot;NIES PCA Eigenvalue&quot;,
        xlab = &quot;Principal Components&quot;,
        ylab = &quot;Eigenvalue&quot;,
        col =&quot;lightskyblue2&quot;)
lines(x = 1:nrow(nies_geno_eigenval), nies_geno_eigenval$V1, 
      type = &quot;b&quot;, pch = 19, col = &quot;red&quot;)</code></pre>
<p><img src="/posts/2018-08-19-using-gaston-for-estimating-heritability-1_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p>There are 112 PCs with an eigenvalue &gt;1.</p>
</div>
<div id="load-eigenvector-files" class="section level3">
<h3>10. Load eigenvector files</h3>
<pre class="r"><code>nies_geno_eigenvec &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/nies_geno_pca.eigenvec&#39;, header = F)</code></pre>
</div>
<div id="generate-eigenvector-plots" class="section level3">
<h3>11. Generate eigenvector plots</h3>
<pre class="r"><code>plot(nies_geno_eigenvec$V3, nies_geno_eigenvec$V4, xlab = &quot;PC1&quot;, ylab = &quot;PC2&quot;, main = &quot;PC1 vs PC2 eigenvectors&quot;) </code></pre>
<p><img src="/posts/2018-08-19-using-gaston-for-estimating-heritability-1_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre class="r"><code>plot(nies_geno_eigenvec$V4, nies_geno_eigenvec$V5, xlab = &quot;PC2&quot;, ylab = &quot;PC3&quot;, main = &quot;PC2 vs PC3 eigenvectors&quot;)</code></pre>
<p><img src="/posts/2018-08-19-using-gaston-for-estimating-heritability-1_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p>The outlier individual is still present in these plots. I need to identify and remove the individual from the data set. Rod speculated that this outlier may have been caused by a technical error. Since the individuals sequenced are part of the core pedigree, there should only be a single cluster of individuals in these plots. Therefore, having an individual that is significantly further away from the cluster is unusual.</p>
<div id="a.-add-labels-to-the-plot-to-identify-the-outlier" class="section level4">
<h4>11a. Add labels to the plot to identify the outlier</h4>
<pre class="r"><code>require(ggplot2)</code></pre>
<pre><code>## Loading required package: ggplot2</code></pre>
<pre class="r"><code>pc1Vpc2 &lt;- data.frame(PC1 = nies_geno_eigenvec$V3, PC2 = nies_geno_eigenvec$V4, names = nies_geno_eigenvec$V2)
ggplot(pc1Vpc2, aes(PC1, PC2)) + geom_point() + geom_text(aes(label=names))</code></pre>
<p><img src="/posts/2018-08-19-using-gaston-for-estimating-heritability-1_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<p>The outlier’s UUID is 273431.</p>
</div>
</div>
<div id="remove-outlier-individual-from-phenotypic-and-genomic-data-sets." class="section level3">
<h3>12. Remove outlier individual from phenotypic and genomic data sets.</h3>
<p>Check that the individuals are still in the same order after removing the data set.</p>
<ul>
<li>Removed UUID 273431 from the ID list</li>
</ul>
<p>Remove individual from genomic data set</p>
<pre class="bash"><code>plink1.9 --bfile merged_nies_geno --keep filtered_pheno_uuid3 --make-bed --out merged_nies_geno_210818</code></pre>
<pre class="r"><code>merged_nies_fam_210818 &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/merged_nies/merged_nies_geno_210818.fam&#39;, header = F)
head(merged_nies_fam_210818)</code></pre>
<pre><code>##   V1     V2 V3 V4 V5 V6
## 1  1 219960  0  0  1 -9
## 2  1 313180  0  0  1 -9
## 3  1 320511  0  0  2 -9
## 4  1 400011  0  0  1 -9
## 5  1 400013  0  0  1 -9
## 6  1 316131  0  0  2 -9</code></pre>
<pre class="r"><code>geno_data_uuid2 &lt;- merged_nies_fam_210818$V2</code></pre>
<p>Remove individual from phenotypic data set</p>
<pre class="r"><code>nies_pheno_210818 &lt;- nies_pheno[nies_pheno$UUID %in% c(geno_data_uuid2), ]
head(nies_pheno_210818)</code></pre>
<pre><code>##      UUID R.Logmar.VA L.Logmar.VA R.Sph..pre.dilate. R.Cyl..pre.dilate.
## 1  219960        0.02       -0.04               0.25               0.00
## 2  313180        0.10        0.16               0.00              -0.75
## 7  320511       -0.06        0.02               1.00               0.00
## 11 400011       -0.14       -0.10               0.75              -0.25
## 13 400013        0.30        0.30               1.50              -0.25
## 14 316131        0.08        0.26               2.00              -1.00
##    R.Axis..pre.dilate. L.Sph..pre.dilate. L.Cyl..pre.dilate.
## 1                    0               0.25              -0.50
## 2                   28              -0.50              -0.25
## 7                    0               1.25              -0.25
## 11                 157               1.25              -0.50
## 13                  82               1.75              -0.50
## 14                  69               1.00              -1.00
##    L.Axis..pre.dilate. R.K.value.H R.K.Value.H.Axis R.K.value.V
## 1                   79       42.00                2       43.00
## 2                  164       41.25                7       42.25
## 7                   29       42.50              177       43.50
## 11                  20       42.50              165       43.25
## 13                 100       43.00                0       43.00
## 14                 114       42.75               63       44.00
##    R.K.value.V.Axis L.K.value.H L.K.value.H.Axis L.K.value.V
## 1                92       42.50                5       43.50
## 2                97       41.50              168       42.00
## 7                87       42.00              170       42.50
## 11               75       42.75               17       43.25
## 13               90       42.75              111       43.00
## 14              153       43.00              115       43.50
##    L.K.value.V.Axis R.Pachimetry L.Pachimetry R.Axial.Length
## 1                95          532          554          24.31
## 2                78          608          612          25.02
## 7                80          499          512          23.37
## 11              107          602          601          23.32
## 13               21          552          550          22.93
## 14               25          583          588          23.49
##    L.Axial.Length R.AC.Depth L.AC.Depth R.IOP.mmHg L.IOP.mmHg R.CDR L.CDR
## 1           24.10       3.09       3.03         14         14   0.9   0.9
## 2           25.21       3.38       3.92         16         15   0.9   0.7
## 7           23.65       3.29       3.08         17         18   0.4   0.4
## 11          23.34       3.81       3.83         15         17   0.6   0.6
## 13          22.70       3.22       3.13         18         18   0.4   0.4
## 14          23.54       3.24       3.04         12         12   0.3   0.3
##    totaluvafmm
## 1          0.0
## 2         74.4
## 7         27.5
## 11        47.0
## 13        20.1
## 14        12.9</code></pre>
</div>
<div id="re-run-pca-on-genomic-data-set" class="section level3">
<h3>13. Re-run PCA on genomic data set</h3>
<pre class="bash"><code>plink1.9 --bfile merged_nies_pheno_210818 --pca 120 var-wts --out plink_output/nies_pca_210818</code></pre>
</div>
<div id="load-eigenvalue-file-and-generate-screeplot" class="section level3">
<h3>14. Load eigenvalue file and generate screeplot</h3>
<pre class="r"><code>nies_eigenval_210818 &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/nies_pca_210818.eigenval&#39;, header = F)

barplot(nies_eigenval_210818$V1, 
        names.arg = 1:nrow(nies_eigenval_210818), 
        main = &quot;NIES PCA Eigenvalue&quot;,
        xlab = &quot;Principal Components&quot;,
        ylab = &quot;Eigenvalue&quot;,
        col =&quot;lightskyblue2&quot;)
lines(x = 1:nrow(nies_eigenval_210818), nies_eigenval_210818$V1, 
      type = &quot;b&quot;, pch = 19, col = &quot;red&quot;)</code></pre>
<p><img src="/posts/2018-08-19-using-gaston-for-estimating-heritability-1_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
</div>
<div id="load-eigenvector-file-and-check-that-outlier-is-not-present-in-pca-plots" class="section level3">
<h3>16. Load eigenvector file and check that outlier is not present in PCA plots</h3>
<pre class="r"><code>nies_eigenvec_210818 &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/nies_pca_210818.eigenvec&#39;, header = F)

plot(nies_eigenvec_210818$V3, nies_eigenvec_210818$V4, xlab = &quot;PC1&quot;, ylab = &quot;PC2&quot;, main = &quot;PC1 vs PC2 eigenvectors&quot;)</code></pre>
<p><img src="/posts/2018-08-19-using-gaston-for-estimating-heritability-1_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
<div id="load-genomic-data-1" class="section level3">
<h3>17. Load genomic data</h3>
<pre class="r"><code>require(gaston)</code></pre>
<pre><code>## Loading required package: gaston</code></pre>
<pre><code>## Loading required package: Rcpp</code></pre>
<pre><code>## Loading required package: RcppParallel</code></pre>
<pre><code>## 
## Attaching package: &#39;RcppParallel&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Rcpp&#39;:
## 
##     LdFlags</code></pre>
<pre><code>## Gaston set number of threads to 2. Use setThreadOptions() to modify this.</code></pre>
<pre><code>## 
## Attaching package: &#39;gaston&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     sigma</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     cbind, rbind</code></pre>
<pre class="r"><code>merged_nies_210818 &lt;- read.bed.matrix(&quot;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/merged_nies/merged_nies_geno_210818.bed&quot;)</code></pre>
<pre><code>## Reading C:/Users/Martha/Documents/Honours/Project/honours.project/Data/merged_nies/merged_nies_geno_210818.fam 
## Reading C:/Users/Martha/Documents/Honours/Project/honours.project/Data/merged_nies/merged_nies_geno_210818.bim 
## Reading C:/Users/Martha/Documents/Honours/Project/honours.project/Data/merged_nies/merged_nies_geno_210818.bed 
## ped stats and snps stats have been set. 
## &#39;p&#39; has been set. 
## &#39;mu&#39; and &#39;sigma&#39; have been set.</code></pre>
</div>
<div id="generate-grm-using-gaston" class="section level3">
<h3>18. Generate GRM using gaston</h3>
<p>Ensure that the individuals are in the same order as the phenotypic and genomic data.</p>
<pre class="r"><code>merged_nies_GRM &lt;- GRM(merged_nies_210818)</code></pre>
</div>
<div id="compute-eigen-decomposition" class="section level3">
<h3>19. Compute eigen decomposition</h3>
<pre class="r"><code>merged_nies_eiK &lt;- eigen(merged_nies_GRM)</code></pre>
</div>
<div id="deal-with-small-negative-eigenvalues" class="section level3">
<h3>20. Deal with small negative eigenvalues</h3>
<pre class="r"><code>merged_nies_eiK$values[ merged_nies_eiK$values &lt; 0] &lt;- 0</code></pre>
</div>
<div id="perform-pca-on-grm" class="section level3">
<h3>18. Perform PCA on GRM</h3>
<pre class="r"><code>merged_nies_PC &lt;- sweep(merged_nies_eiK$vectors, 2, sqrt(merged_nies_eiK$values), &quot;*&quot;)</code></pre>
</div>
<div id="plot-eigenvector-values-from-pca-of-the-grm" class="section level3">
<h3>19. Plot eigenvector values from PCA of the GRM</h3>
<pre class="r"><code>plot(merged_nies_PC[,1], merged_nies_PC[,2], xlab = &quot;PC1&quot;, ylab = &quot;PC2&quot;, main = &quot;GRM PC1 vs PC2&quot;) </code></pre>
<p><img src="/posts/2018-08-19-using-gaston-for-estimating-heritability-1_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
</div>
<div id="test-lmm.diago-function-gaston" class="section level3">
<h3>20. Test lmm.diago() function (gaston)</h3>
<pre class="r"><code>Y &lt;- nies_pheno_210818$R.K.value.H
fit &lt;- lmm.diago(Y, eigenK = merged_nies_eiK)</code></pre>
<pre><code>## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 41.4731
## [Iteration 2] Current point = 0.231523 df = 18.3374
## [Iteration 3] Current point = 0.518644 df = 0.833621
## [Iteration 4] Current point = 0.530343 df = -0.0125581
## [Iteration 5] Current point = 0.530172 df = -2.86236e-006</code></pre>
<p>h^2 of horizontal R K-value = 53.01% (?)</p>
</div>
<div id="repeat-for-all-phenotypic-variables" class="section level3">
<h3>21. Repeat for all phenotypic variables</h3>
<pre class="r"><code>for (i in 2:ncol(nies_pheno_210818)){
  fit &lt;- lmm.diago(nies_pheno_210818[,i], eigenK = merged_nies_eiK)}</code></pre>
<pre><code>## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = -2.2559
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = -16.655
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = -9.57183
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 28.3745
## [Iteration 2] Current point = 0.484116 df = 4.41849
## [Iteration 3] Current point = 0.550692 df = -0.406865
## [Iteration 4] Current point = 0.545563 df = -0.00300791
## [Iteration 5] Current point = 0.545525 df = -1.66267e-007
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 2.08593
## [Iteration 2] Current point = 0.0336347 df = 0.117014
## [Iteration 3] Current point = 0.0357434 df = 0.000347961
## [Iteration 4] Current point = 0.0357497 df = 3.06179e-009
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = -10.2621
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = -7.03848
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 7.05039
## [Iteration 2] Current point = 0.0729428 df = 1.33525
## [Iteration 3] Current point = 0.0934138 df = 0.0496894
## [Iteration 4] Current point = 0.0942343 df = 6.77127e-005
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 41.4731
## [Iteration 2] Current point = 0.231523 df = 18.3374
## [Iteration 3] Current point = 0.518644 df = 0.833621
## [Iteration 4] Current point = 0.530343 df = -0.0125581
## [Iteration 5] Current point = 0.530172 df = -2.86236e-006
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = -9.56967
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 49.5036
## [Iteration 2] Current point = 0.266942 df = 26.2542
## [Iteration 3] Current point = 0.821788 df = -9.32438
## [Iteration 4] Current point = 0.775858 df = -1.952
## [Iteration 5] Current point = 0.760616 df = -0.116259
## [Iteration 6] Current point = 0.75959 df = -0.000450576
## [Iteration 7] Current point = 0.759586 df = -6.78389e-009
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = -14.4913
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 35.3151
## [Iteration 2] Current point = 0.1702 df = 13.7743
## [Iteration 3] Current point = 0.334479 df = 1.91493
## [Iteration 4] Current point = 0.363073 df = -0.00294203
## [Iteration 5] Current point = 0.363029 df = -2.35821e-008
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 0.484426
## [Iteration 2] Current point = 0.0064578 df = 0.0122138
## [Iteration 3] Current point = 0.0066291 df = 7.97164e-006
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 47.5036
## [Iteration 2] Current point = 0.21944 df = 22.4879
## [Iteration 3] Current point = 0.552683 df = 3.68487
## [Iteration 4] Current point = 0.614956 df = -0.223631
## [Iteration 5] Current point = 0.61162 df = -0.000894753
## [Iteration 6] Current point = 0.611607 df = -1.43198e-008
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = -5.823
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 38.2559
## [Iteration 2] Current point = 0.27791 df = 16.0916
## [Iteration 3] Current point = 0.570976 df = 0.713281
## [Iteration 4] Current point = 0.583485 df = -0.00665774
## [Iteration 5] Current point = 0.58337 df = -5.94581e-007
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 40.2764
## [Iteration 2] Current point = 0.310726 df = 16.9307
## [Iteration 3] Current point = 0.640107 df = 0.435845
## [Iteration 4] Current point = 0.647777 df = -0.00294086
## [Iteration 5] Current point = 0.647726 df = -1.34981e-007
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 23.1767
## [Iteration 2] Current point = 0.197614 df = 7.99998
## [Iteration 3] Current point = 0.339062 df = 0.598113
## [Iteration 4] Current point = 0.350965 df = 7.42912e-005
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 23.1579
## [Iteration 2] Current point = 0.183926 df = 8.23487
## [Iteration 3] Current point = 0.324513 df = 0.719946
## [Iteration 4] Current point = 0.338682 df = 0.000540716
## [Iteration 5] Current point = 0.338693 df = 5.47118e-012
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 27.2665
## [Iteration 2] Current point = 0.18723 df = 8.27687
## [Iteration 3] Current point = 0.296964 df = 0.659801
## [Iteration 4] Current point = 0.307044 df = 0.00182903
## [Iteration 5] Current point = 0.307072 df = 1.20226e-008
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 25.4208
## [Iteration 2] Current point = 0.180651 df = 7.2106
## [Iteration 3] Current point = 0.274918 df = 0.536193
## [Iteration 4] Current point = 0.282972 df = 0.00194078
## [Iteration 5] Current point = 0.283001 df = 2.39764e-008
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 19.0563
## [Iteration 2] Current point = 0.147446 df = 6.12048
## [Iteration 3] Current point = 0.24176 df = 0.503822
## [Iteration 4] Current point = 0.250745 df = 0.00152377
## [Iteration 5] Current point = 0.250772 df = 1.21091e-008
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 19.565
## [Iteration 2] Current point = 0.131325 df = 5.87129
## [Iteration 3] Current point = 0.206538 df = 0.500194
## [Iteration 4] Current point = 0.214074 df = 0.00268863
## [Iteration 5] Current point = 0.214115 df = 7.4301e-008
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 72.9477
## [Iteration 2] Current point = 0.158165 df = 35.0727
## [Iteration 3] Current point = 0.434828 df = 12.9756
## [Iteration 4] Current point = 0.647277 df = -1.20794
## [Iteration 5] Current point = 0.632241 df = -0.0237355
## [Iteration 6] Current point = 0.631934 df = -9.17477e-006
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 75.6479
## [Iteration 2] Current point = 0.147316 df = 35.2271
## [Iteration 3] Current point = 0.385568 df = 13.2911
## [Iteration 4] Current point = 0.589986 df = 0.309687
## [Iteration 5] Current point = 0.5945 df = -0.00121467
## [Iteration 6] Current point = 0.594482 df = -1.89248e-008
## Optimization in interval [0, 1]
## Optimizing with p = 0
## [Iteration 1] Current point = 0 df = 24.1626
## [Iteration 2] Current point = 0.244014 df = 8.88222
## [Iteration 3] Current point = 0.432413 df = 0.131632
## [Iteration 4] Current point = 0.435111 df = -0.000213949</code></pre>
</div>
<div id="include-two-principal-components-for-estimating-heritability" class="section level3">
<h3>22. Include two principal components for estimating heritability</h3>
<pre class="r"><code>for (i in 2:ncol(nies_pheno_210818)){
  fit &lt;- lmm.diago(nies_pheno_210818[,i], eigenK = merged_nies_eiK, p = 2)}</code></pre>
<pre><code>## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = -7.11013
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = -13.3561
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = -8.19357
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 28.7103
## [Iteration 2] Current point = 0.599149 df = -3.46492
## [Iteration 3] Current point = 0.561391 df = -0.198617
## [Iteration 4] Current point = 0.558957 df = -0.000713875
## [Iteration 5] Current point = 0.558948 df = -9.27375e-009
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 3.78121
## [Iteration 2] Current point = 0.0614073 df = 0.309897
## [Iteration 3] Current point = 0.0673109 df = 0.00175775
## [Iteration 4] Current point = 0.0673448 df = 5.52169e-008
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = -10.5435
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = -4.78203
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 6.30518
## [Iteration 2] Current point = 0.0910141 df = 0.794675
## [Iteration 3] Current point = 0.105783 df = 0.011704
## [Iteration 4] Current point = 0.106007 df = 2.46312e-006
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 29.269
## [Iteration 2] Current point = 0.347647 df = 9.62902
## [Iteration 3] Current point = 0.542008 df = -1.62963
## [Iteration 4] Current point = 0.519432 df = -0.0548232
## [Iteration 5] Current point = 0.518619 df = -6.32935e-005
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = -7.36338
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 34.8249
## [Iteration 2] Current point = 0.531217 df = 14.0813
## [Iteration 3] Current point = 0.879187 df = -27.5399
## [Iteration 4] Current point = 0.823716 df = -9.91648
## [Iteration 5] Current point = 0.775494 df = -2.20567
## [Iteration 6] Current point = 0.757915 df = -0.152536
## [Iteration 7] Current point = 0.756513 df = -0.000811872
## [Iteration 8] Current point = 0.756506 df = -2.31775e-008
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = -11.3499
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 14.3799
## [Iteration 2] Current point = 0.263585 df = 1.54924
## [Iteration 3] Current point = 0.295381 df = -0.0212347
## [Iteration 4] Current point = 0.294957 df = -4.56719e-006
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = -0.187161
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 29.0126
## [Iteration 2] Current point = 0.435271 df = 8.12837
## [Iteration 3] Current point = 0.613725 df = -1.21958
## [Iteration 4] Current point = 0.59433 df = -0.0309852
## [Iteration 5] Current point = 0.593811 df = -2.01423e-005
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = -2.94506
## [Iteration 1] maximum at min = 0
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 36.5065
## [Iteration 2] Current point = 0.356508 df = 12.8279
## [Iteration 3] Current point = 0.610013 df = -0.571562
## [Iteration 4] Current point = 0.600536 df = -0.0047655
## [Iteration 5] Current point = 0.600456 df = -3.27571e-007
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 40.347
## [Iteration 2] Current point = 0.363064 df = 15.4634
## [Iteration 3] Current point = 0.677099 df = -0.587165
## [Iteration 4] Current point = 0.667423 df = -0.00565348
## [Iteration 5] Current point = 0.667328 df = -5.21474e-007
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 21.8958
## [Iteration 2] Current point = 0.254171 df = 5.84146
## [Iteration 3] Current point = 0.3687 df = 0.125045
## [Iteration 4] Current point = 0.37121 df = -5.02903e-005
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 20.5946
## [Iteration 2] Current point = 0.243888 df = 5.45003
## [Iteration 3] Current point = 0.351788 df = 0.10615
## [Iteration 4] Current point = 0.353935 df = -3.52901e-005
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 25.019
## [Iteration 2] Current point = 0.208385 df = 6.76504
## [Iteration 3] Current point = 0.307491 df = 0.337584
## [Iteration 4] Current point = 0.312871 df = 0.000239402
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 24.4286
## [Iteration 2] Current point = 0.198858 df = 6.34809
## [Iteration 3] Current point = 0.288122 df = 0.349883
## [Iteration 4] Current point = 0.293561 df = 0.000614758
## [Iteration 5] Current point = 0.293571 df = 1.79782e-009
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 17.157
## [Iteration 2] Current point = 0.18723 df = 4.28128
## [Iteration 3] Current point = 0.263656 df = 0.122049
## [Iteration 4] Current point = 0.265935 df = 1.29855e-005
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 15.7871
## [Iteration 2] Current point = 0.164015 df = 3.26658
## [Iteration 3] Current point = 0.21526 df = 0.0952811
## [Iteration 4] Current point = 0.216838 df = 5.61709e-005
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 33.0992
## [Iteration 2] Current point = 0.327339 df = 13.6128
## [Iteration 3] Current point = 0.632864 df = -2.41685
## [Iteration 4] Current point = 0.598798 df = -0.117288
## [Iteration 5] Current point = 0.596974 df = -0.000289291
## [Iteration 6] Current point = 0.596969 df = -1.76379e-009
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 30.6574
## [Iteration 2] Current point = 0.2758 df = 12.4104
## [Iteration 3] Current point = 0.5439 df = -0.185709
## [Iteration 4] Current point = 0.540475 df = -0.000632608
## [Iteration 5] Current point = 0.540463 df = -7.32253e-009
## Optimization in interval [0, 1]
## Optimizing with p = 2
## [Iteration 1] Current point = 0 df = 24.2256
## [Iteration 2] Current point = 0.303903 df = 7.32815
## [Iteration 3] Current point = 0.464196 df = -0.252431
## [Iteration 4] Current point = 0.459249 df = -0.000927276
## [Iteration 5] Current point = 0.459231 df = -1.23407e-008</code></pre>
<p>The output of this last step contains the narrow sense heritability estimates for each of the 27 variables in the phenotypic data set and the corresponding degrees of freedom. (need to add labels but it is in order of the variables as listed in the spreadsheet)</p>
<p>When interpreting the results, does the last iteration indicate the heritability estimate? Why are there varying numbers of iterations for the variables? For the variables where there is only 1 iteration and it says maximum at min = 0, how is this interpreted? How do I estimate the heritability for the principal components?</p>
</div>
</div>

      </div>
      
      <div class="post-content">
        <div id="disqus_thread"></div>
<script>
(function() {
var d = document, s = d.createElement('script');
s.src = 'https://http-martha-labbook-netlify-com.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>

      </div>
      
    </div>
  </div>
  <div class="pure-u-1-24 pure-u-md-1-6"></div>
</div>


<div class="footer pure-g">
  <div class="pure-u-1-24 pure-u-md-1-6"></div>
  <div class="pure-u-11-12 pure-u-md-2-3">
    <div class="footer-content">
      <div class="pure-menu pure-menu-horizontal">
        <ul>
          
          <li class="pure-menu-item" id="foot-name">© Martha Aquino</li>
          

          
          <li class="pure-menu-item">
            <a href="mailto:eunise.aquino@connect.qut.edu.au" class="pure-menu-link">Email</a>
          </li>
          

          
          <li class="pure-menu-item">
            <a href="https://github.com/https://github.com/marthaqut/electronic_lab_notebook" class="pure-menu-link">GitHub</a>
          </li>
          

          

          

        </ul>
        <a href="#" class="pure-menu-link pull-right" id="gototop-btn">↑↑</a>
        
      </div>
      
      <p id="foot-copyright">(c) 2018 -- All rights reserved.</p>
      
    </div>
  </div>
  <div class="pure-u-1-24 pure-u-md-1-6"></div>
</div>



  <script src="/highlight/highlight.min.js"></script>
  
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/go.min.js"></script>
  
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/vim.min.js"></script>
  
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/shell.min.js"></script>
  
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/bash.min.js"></script>
  
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/markdown.min.js"></script>
  
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/r.min.js"></script>
  
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/languages/python.min.js"></script>
   
  <script>hljs.initHighlighting();</script>


<script type="text/javascript">
  go();
  window.addEventListener('resize', go);
  function go(){
    var d=document.querySelector(".desktop")
    if(document.documentElement.clientWidth > 768) {
      d.style.display="";
    } else {
      d.style.display="none";
    }
  }
</script>
<script type="text/javascript">
  var a=document.querySelector("#toggle-btn"),b=document.querySelector("#toggle-content");a.addEventListener("click",function(){""==b.style.display?(b.style.display="none",a.innerHTML="☰"):(b.style.display="",a.innerHTML="X")});</script>


</body>
</html>



