---
title: Performing GWAS on heritable traits
author: Martha Aquino
date: '2018-09-07'
slug: performing-gwas-on-heritable-traits
categories:
  - Coding
  - Experiments
tags:
  - Gaston
  - GCTA
  - Genomic Data
  - GWAS
  - Heritability
---

```{r}
require(magrittr)
require(dplyr)
require(gaston)
require(qqman)
```


#Introduction
After running GWAS parallel in GCTA and gaston, we were concerned that GCTA was over burdening the test because the program was developed to deal with very large amounts of data. Considering our sample size was very small, random effects that may be in the GCTA models could have very significant effects on the GWAS results. This was evident on the manhattan plots generated for the same data, with the same adjustments between GCTA and gaston. In light of this, we have decided to move the GWAS analysis back to gaston. 

I will be performing GWAS on 21 heritable traits (16 individual phenotypes and 5 principal components). I will be using the likelihood ratio test (LRT) as opposed to the wald test and score test as it is the most commonly used in GWAS and is suitable for smaller sample sizes (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4547377/). I will also be correcting for any effects of population stratification by including two prinicipal components as I have done with the heritability estimates. 

For the Norfolk Island population, we set a genome-wide signficance threshold of p < 1.84e-7 rather than the general p < 5e-8 because that was calculated specifically for the sample size in the study.

#Methods and Results 

###1. Load phenotypic data
```{r}
nies_pheno_020918 <- read.csv('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/nies_pheno_020918.csv', header = T)
head(nies_pheno_020918)
```

###2. Load genomic data (bed/bim/fam)
```{r}
merged_nies_210818 <- read.bed.matrix("C:/Users/Martha/Documents/Honours/Project/honours.project/Data/merged_nies/merged_nies_geno_210818.bed")
```

###3. Generate GRM, eigenK, and principal components
```{r}
merged_nies_GRM <- GRM(merged_nies_210818)
merged_nies_eiK <- eigen(merged_nies_GRM)
merged_nies_eiK$values[ merged_nies_eiK$values < 0] <- 0
merged_nies_PC <- sweep(merged_nies_eiK$vectors, 2, sqrt(merged_nies_eiK$values), "*")
```

The genetic relationship matrix (GRM) will be used to account for the degree of relatedness among individuals, and the principal components will be used to correct for hiden population structure. Both of these ensure that any associations detected are not due to relatedness or hidden structures in the data i.e. they are true phenotypic-variant associations. 

###4. Perform GWAS on R K-value V (most heritable trait)
```{r}
r_kvalV_gwas <- association.test(merged_nies_210818, nies_pheno_020918$R.K.value.V, method="lmm", test = "lrt", K = merged_nies_GRM, eigenK = merged_nies_eiK, p = 2)
r_kvalV_gwas <- na.omit(r_kvalV_gwas)
r_kvalV_gwas_filtered <- r_kvalV_gwas %>% filter(-log10(p)>1) %>% filter(freqA2 < 0.99)
qqman::manhattan(x = r_kvalV_gwas_filtered, chr = "chr", bp = "pos", p = "p", snp = "id", ylim = c(0,10), genomewideline = -log10(1.84e-7), main = "R K-value V", annotatePval = 1.84e-7)
```

###5. Repeat GWAS on other heritable traits 
```{r}
nies_heritable_pheno <- read.csv('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/nies_heritable_pheno.csv', header = T)
head(nies_heritable_pheno)
```

I compiled the data for the heritable traits into a new spreadsheet (loaded above). 

```{r}
pheno_gwas <- NULL
for (i in c(2:ncol(nies_heritable_pheno))){
  pheno_colnames <- colnames(nies_heritable_pheno[i])
  her_pheno_gwas <- association.test(merged_nies_210818, nies_heritable_pheno[,i], method = "lmm", 
                                     test = "lrt", K = merged_nies_GRM, eigenK = merged_nies_eiK, p =2)
  her_pheno_gwas <- na.omit(her_pheno_gwas)
  pheno_gwas_filtered <- her_pheno_gwas %>% filter(-log10(p)>1) %>% filter(freqA2 < 0.99)
  tiff(paste('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/gwas_plot_', pheno_colnames, '_', '.png'), width = 800, height = 500, res = 150, compression = 'lzw')
  qqman::manhattan(x = pheno_gwas_filtered, chr = "chr", bp = "pos", p = "p", snp = "id", genomewideline = -log10(1.84e-7), main = paste(pheno_colnames))
  dev.off()
  pheno_gwas <- pheno_gwas_filtered
}
```

The loop above generates manhattan plots for the GWAS results of all 21 heritable traits. 

###6. Annotate manhattan plots with significant hits 

I tried adding an annotation argument to the loop above to annotate hits that passed the genome-wide significance threshold. However, since not all manhattan plots had significant hits, the argument caused an error. I removed the argument from the loop and it was successful in generating all 21 manhattan plots. Of these, approximately 9 plots have significant hits and therefore require annotation. 

```{r}
###uvaf
uvaf_gwas <- association.test(merged_nies_210818, nies_heritable_pheno$totaluvafmm, method="lmm", test = "lrt", K = merged_nies_GRM, eigenK = merged_nies_eiK, p = 2)
uvaf_gwas <- na.omit(uvaf_gwas)
uvaf_gwas_filtered <- uvaf_gwas %>% filter(-log10(p)>1) %>% filter(freqA2 < 0.99)
manhattan(x = uvaf_gwas_filtered, chr = "chr", bp = "pos", p = "p", snp = "id", ylim = c(0,10), genomewideline = -log10(1.84e-7), main = "Ultraviolet Autofluorescence", annotatePval = 1.84e-7)

###PC3 
pc3_gwas <- association.test(merged_nies_210818, nies_heritable_pheno$coord.Dim.3, method="lmm", test = "lrt", K = merged_nies_GRM, eigenK = merged_nies_eiK, p = 2)
pc3_gwas <- na.omit(pc3_gwas)
pc3_gwas_filtered <- pc3_gwas %>% filter(-log10(p)>1) %>% filter(freqA2 < 0.99)
manhattan(x = pc3_gwas_filtered, chr = "chr", bp = "pos", p = "p", snp = "id", ylim = c(0,10), genomewideline = -log10(1.84e-7), main = "Component 3", annotatePval = 1.84e-7)

###PC8
pc8_gwas <- association.test(merged_nies_210818, nies_heritable_pheno$coord.Dim.8, method="lmm", test = "lrt", K = merged_nies_GRM, eigenK = merged_nies_eiK, p = 2)
pc8_gwas <- na.omit(pc8_gwas)
pc8_gwas_filtered <- pc8_gwas %>% filter(-log10(p)>1) %>% filter(freqA2 < 0.99)
manhattan(x = pc8_gwas_filtered, chr = "chr", bp = "pos", p = "p", snp = "id", ylim = c(0,10), genomewideline = -log10(1.84e-7), main = "Component 8", annotatePval = 1.84e-7)

###L ACD
l_acd_gwas <- association.test(merged_nies_210818, nies_heritable_pheno$L.AC.Depth, method="lmm", test = "lrt", K = merged_nies_GRM, eigenK = merged_nies_eiK, p = 2)
l_acd_gwas <- na.omit(l_acd_gwas)
l_acd_gwas_filtered <- l_acd_gwas %>% filter(-log10(p)>1) %>% filter(freqA2 < 0.99)
manhattan(x = l_acd_gwas_filtered, chr = "chr", bp = "pos", p = "p", snp = "id", ylim = c(0,10), genomewideline = -log10(1.84e-7), main = "Left Anterior Chamber Depth", annotatePval = 1.84e-7)

###L CDR
l_cdr_gwas <- association.test(merged_nies_210818, nies_heritable_pheno$L.CDR, method="lmm", test = "lrt", K = merged_nies_GRM, eigenK = merged_nies_eiK, p = 2)
l_cdr_gwas <- na.omit(l_cdr_gwas)
l_cdr_gwas_filtered <- l_cdr_gwas %>% filter(-log10(p)>1) %>% filter(freqA2 < 0.99)
manhattan(x = l_cdr_gwas_filtered, chr = "chr", bp = "pos", p = "p", snp = "id", ylim = c(0,10), genomewideline = -log10(1.84e-7), main = "Left Cup-to-Disc Ratio", annotatePval = 1.84e-7)

###L Kval H
l_KvalH_gwas <- association.test(merged_nies_210818, nies_heritable_pheno$L.K.value.H, method="lmm", test = "lrt", K = merged_nies_GRM, eigenK = merged_nies_eiK, p = 2)
l_KvalH_gwas <- na.omit(l_KvalH_gwas)
l_KvalH_gwas_filtered <- l_KvalH_gwas %>% filter(-log10(p)>1) %>% filter(freqA2 < 0.99)
manhattan(x = l_KvalH_gwas_filtered, chr = "chr", bp = "pos", p = "p", snp = "id", ylim = c(0,10), genomewideline = -log10(1.84e-7), main = "Left Horizontal K-value", annotatePval = 1.84e-7)

###R ACD 
r_acd_gwas <- association.test(merged_nies_210818, nies_heritable_pheno$R.AC.Depth, method="lmm", test = "lrt", K = merged_nies_GRM, eigenK = merged_nies_eiK, p = 2)
r_acd_gwas <- na.omit(r_acd_gwas)
r_acd_gwas_filtered <- r_acd_gwas %>% filter(-log10(p)>1) %>% filter(freqA2 < 0.99)
manhattan(x = r_acd_gwas_filtered, chr = "chr", bp = "pos", p = "p", snp = "id", ylim = c(0,10), genomewideline = -log10(1.84e-7), main = "Right Anterior Chamber Depth", annotatePval = 1.84e-7)

###R Cyl
r_cyl_gwas <- association.test(merged_nies_210818, nies_heritable_pheno$R.Cyl..pre.dilate, method="lmm", test = "lrt", K = merged_nies_GRM, eigenK = merged_nies_eiK, p = 2)
r_cyl_gwas <- na.omit(r_cyl_gwas)
r_cyl_gwas_filtered <- r_cyl_gwas %>% filter(-log10(p)>1) %>% filter(freqA2 < 0.99)
manhattan(x = r_cyl_gwas_filtered, chr = "chr", bp = "pos", p = "p", snp = "id", ylim = c(0,15), genomewideline = -log10(1.84e-7), main = "Right Cylindrical Pre-Dilate Refractive Error", annotatePval = 1.84e-7)

```


