---
title: Generating stats for merged data and extracting data for NIES
author: Martha Aquino
date: '2018-08-03'
slug: generating-stats-for-merged-data-and-extracting-data-for-nies
categories:
  - Coding
  - Experiments
tags:
  - Data Cleaning
  - Genomic Data
  - Linux
  - PLINK
---



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Now that the merge has been successful and there should no longer be duplicated individuals, general stats need to be performed on the merged data set. Subsequently, the final step to this data prep and cleaning is to extract data for the NIES individuals only.</p>
<div id="methods-and-results" class="section level2">
<h2>Methods and Results</h2>
<p>Note: Most processes up until the merge from the previous entry was performed on taurus. I transferred filtered_merge to local directory. These stat tests will be performed locally. ###1. Generate allele freq stats</p>
<pre class="bash"><code>plink --bfile plink_output/filtered_merge --freq --out plink_output/merge_allele_freq</code></pre>
<div id="load-allele-freq-file" class="section level3">
<h3>2. Load allele freq file</h3>
<pre class="r"><code>merged_allele_freq &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_allele_freq.frq&#39;, header = T)

head(merged_allele_freq)</code></pre>
<pre><code>##   CHR         SNP A1 A2      MAF NCHROBS
## 1   1  rs62637813  C  G 0.125000     104
## 2   1 rs146477069  G  A 0.009804     102
## 3   1 rs141149254  A  G 0.031250      64
## 4   1   rs2462492  T  C 0.425900      54
## 5   1 rs143174675  G  T 0.009091     110
## 6   1   rs3091274  C  A 0.000000     110</code></pre>
</div>
<div id="generate-histogram-for-allele-freq-distributions" class="section level3">
<h3>3. Generate histogram for allele freq distributions</h3>
<pre class="r"><code>hist(merged_allele_freq$MAF, main = &quot;Allele Frequency Distributions of Merged Data&quot;, xlab = &quot;Minor Allele Frequency&quot;, xlim = c(0.0, 0.5), col = &quot;plum2&quot;)</code></pre>
<p><img src="/posts/2018-08-03-generating-stats-for-merged-data-and-extracting-data-for-nies_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
</div>
<div id="generate-missingness-statistics" class="section level3">
<h3>4. Generate missingness statistics</h3>
<pre class="bash"><code>plink1.9 --bfile plink_output/filtered_merge --missing --out plink_output/merged_missing</code></pre>
<pre class="r"><code>merged_missing_snp &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_missing.lmiss&#39;, header = T)

head(merged_missing_snp)</code></pre>
<pre><code>##   CHR         SNP N_MISS N_GENO  F_MISS
## 1   1  rs62637813     64    520 0.12310
## 2   1 rs146477069    115    520 0.22120
## 3   1 rs141149254    264    520 0.50770
## 4   1   rs2462492    295    520 0.56730
## 5   1 rs143174675     34    520 0.06538
## 6   1   rs3091274     24    520 0.04615</code></pre>
<pre class="r"><code>merged_missing_indiv &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_missing.imiss&#39;, header = T)

head(merged_missing_indiv)</code></pre>
<pre><code>##   FID    IID MISS_PHENO N_MISS  N_GENO  F_MISS
## 1   1 110050          Y 243446 9155053 0.02659
## 2   1 110110          Y 278192 9155053 0.03039
## 3   1 110130          Y 267987 9155053 0.02927
## 4   1 110150          Y 250349 9155053 0.02735
## 5   1 110160          Y 235054 9155053 0.02567
## 6   1 110320          Y 277136 9155053 0.03027</code></pre>
</div>
<div id="generate-hardy-weinberg-statistics" class="section level3">
<h3>5. Generate hardy-weinberg statistics</h3>
<pre class="bash"><code>plink1.9 --bfile plink_output/filtered_merge --hardy --out plink_output/merged_hardy</code></pre>
<pre class="r"><code>#merged_hardy &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_hardy.hwe&#39;, header = T)

#head(merged_hardy)</code></pre>
</div>
<div id="filter-out-variants-based-on-hwe-p-value" class="section level3">
<h3>6. Filter out variants based on HWE p-value</h3>
<pre class="bash"><code>plink1.9 --bfile plink_output/filtered_merge --hwe 1.8e-7 --make-bed --out plink_output/merged_hwe_filter</code></pre>
<p>Note: unlike previous attempts at generating stats before merging, I did not use a HWE p-val filter before merging this data set. Note: this creates a new file set</p>
<p>Output: - 1897 variants were excluded based on hwe p-val - 9,153,156 variants remain - warning: hwe observation counts vary by more than 10%</p>
</div>
<div id="check-allele-freq-stats-after-filtering" class="section level3">
<h3>7. Check allele freq stats after filtering</h3>
<pre class="bash"><code>plink1.9 --bfile plink_output/merged_hwe_filter --freq --out plink_output/merge_hwefilter_freq</code></pre>
</div>
<div id="load-allele-freq-file-1" class="section level3">
<h3>8. Load allele freq file</h3>
<pre class="r"><code>merged_hwefilter_freq &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merge_hwefilter_freq.frq&#39;, header = T)</code></pre>
</div>
<div id="generate-histogram-of-allele-freq-distributions" class="section level3">
<h3>9. Generate histogram of allele freq distributions</h3>
<pre class="r"><code>hist(merged_hwefilter_freq$MAF, main = &quot;Allele Frequency Distributions of Merged Data after HWE filtering&quot;, xlab = &quot;Minor Allele Frequency&quot;, xlim = c(0.0, 0.5), col = &quot;peachpuff&quot;)</code></pre>
<p><img src="/posts/2018-08-03-generating-stats-for-merged-data-and-extracting-data-for-nies_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
</div>
<div id="load-spreadsheet-with-ni-study-participants" class="section level3">
<h3>10. Load spreadsheet with NI study participants</h3>
<pre class="r"><code>ni_ped_uuid &lt;- read.csv(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/NI_UUID_Ped_20180718.csv&#39;, header = T) 

head(ni_ped_uuid)</code></pre>
<pre><code>##     UUID Ped_ID LAB_ID  PID  SID NIES KCCGID Gender        DOB IN_2000
## 1 110020     NA   1002 &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;   &lt;NA&gt; Female  5/16/1929       1
## 2 110030     NA   1003 &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;   &lt;NA&gt; Female  5/23/1928       1
## 3 110040   3676   1004 &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;   &lt;NA&gt; Female  2/08/1959       1
## 4 110050    765   1005 &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;   &lt;NA&gt;   Male 11/07/1954       1
## 5 110080     NA   1008 &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;   &lt;NA&gt; Female 10/07/1933       1
## 6 110090     NA   1009 &lt;NA&gt; &lt;NA&gt; &lt;NA&gt;   &lt;NA&gt; Female 10/08/1968       1
##   AGE_2000NIHS IN_2010 AGE_2010NIHS IN_NIES AGE_NIES RECOLLECT COREPED_1
## 1           71       0           80       0       79         0         0
## 2           72       0           81       0       80         0         0
## 3           41       0           50       0       49         0         0
## 4           46       0           55       0       54         0         1
## 5           67       0           76       0       75         0         0
## 6           32       0           41       0       40         0         0
##   COREPED_2 Miles_core DNA_NIHS X2000NIHS_GWAS X2010NIHS_GWAS GWAS_NIHS
## 1         0          0        1              0              0         0
## 2         0          0        1              0              0         0
## 3         0          0        1              0              0         0
## 4         1          1        1              1              0         1
## 5         0          0        1              0              0         0
## 6         0          0        1              0              0         0
##   DNA_NIES GWAS_NIES Total_GWAS NI_exprs WGS_data DNA_TOTAL
## 1        0         0          0        0        0         1
## 2        0         0          0        0        0         1
## 3        0         0          0        0        0         1
## 4        0         0          1        0        0         1
## 5        0         0          0        0        0         1
## 6        0         0          0        0        0         1
##   Relation.to.NIHS.Pedigree..n.6379. patID matID Ped_Gender isParent
## 1                               &lt;NA&gt;    NA    NA       &lt;NA&gt;       NA
## 2                               &lt;NA&gt;    NA    NA       &lt;NA&gt;       NA
## 3                               &lt;NA&gt;    NA    NA       &lt;NA&gt;       NA
## 4                               &lt;NA&gt;   764   762       Male    FALSE
## 5                               &lt;NA&gt;    NA    NA       &lt;NA&gt;       NA
## 6                               &lt;NA&gt;    NA    NA       &lt;NA&gt;       NA</code></pre>
</div>
<div id="isolate-all-nies-individuals-using-nies-column-and-all-nies-indivs-with-sequencing-data-using-gwas_nies-column" class="section level3">
<h3>11. Isolate all NIES individuals using “NIES” column, and all NIES indivs with sequencing data using “GWAS_NIES” column</h3>
<pre class="r"><code>gwas_niesID &lt;- read.csv(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/gwas_niesID.csv&#39;, header = F)

head(gwas_niesID)</code></pre>
<pre><code>##       V1
## 1 312481
## 2 311020
## 3 255711
## 4 321130
## 5 232110
## 6 253810</code></pre>
<p>There are 363 NIES individuals that may have sequencing data identified from the spreadsheet. This list was also converted to .txt to be compatible with PLINK.</p>
</div>
<div id="extract-data-in-plink" class="section level3">
<h3>12. Extract data in PLINK</h3>
<pre class="bash"><code>plink1.9 --bfile plink_output/merged_hwe_filter --keep gwas_niesID.txt --make-bed --out merged_nies</code></pre>
<p>This extracted data for only 361 individuals, which is 2 less than expected. I suspect it may be because of differing paternal and maternal IDs. Therefore, I changed all maternal and paternal IDs in “merged_hwe_filter.fam” to 0s but this did not resolve the issue. PLINK is still only extracting data for 361 individuals (9,153,156 variants; total genotyping rate = 91.68%)</p>
<p>The file sets created in this step will be used for all subsequent analysis.</p>
</div>
<div id="check-allele-frequencies-of-this-final-data-set" class="section level3">
<h3>13. Check allele frequencies of this final data set</h3>
<pre class="bash"><code>plink1.9 --bfile merged_nies --freq --out plink_output/merged_nies_allele_freq</code></pre>
</div>
<div id="load-allele-freq-file-2" class="section level3">
<h3>14. Load allele freq file</h3>
<pre class="r"><code>merged_nies_allele_freq &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_nies_allele_freq.frq&#39;, header = T)

head(merged_nies_allele_freq)</code></pre>
<pre><code>##   CHR         SNP A1 A2      MAF NCHROBS
## 1   1  rs62637813  C  G 0.014330     628
## 2   1 rs146477069  G  A 0.003497     572
## 3   1 rs141149254  A  G 0.063580     346
## 4   1   rs2462492  T  C 0.155400     296
## 5   1 rs143174675  G  T 0.002994     668
## 6   1   rs3091274  C  A 0.000000     690</code></pre>
</div>
<div id="generate-histogram-for-allele-frequency-distributions" class="section level3">
<h3>15. Generate histogram for allele frequency distributions</h3>
<pre class="r"><code>hist(merged_nies_allele_freq$MAF, main = &quot;Allele Frequency Distributions of NIES Data&quot;, xlab = &quot;Minor Allele Frequency&quot;, xlim = c(0.0, 0.5), col = &quot;lightskyblue1&quot;)</code></pre>
<p><img src="/posts/2018-08-03-generating-stats-for-merged-data-and-extracting-data-for-nies_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
</div>
<div id="check-missingness-rates" class="section level3">
<h3>16. Check missingness rates</h3>
<pre class="bash"><code>plink1.9 --bfile merged_nies --missing --out plink_output/merged_nies_missing</code></pre>
<pre class="r"><code>merged_nies_imissing &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_nies_missing.imiss&#39;, header = T)

head(merged_nies_imissing)</code></pre>
<pre><code>##   FID    IID MISS_PHENO  N_MISS  N_GENO  F_MISS
## 1   1 110150          Y  250197 9153156 0.02733
## 2   1 110160          Y  234913 9153156 0.02566
## 3   1 110500          Y  320553 9153156 0.03502
## 4   1 110650          Y  284393 9153156 0.03107
## 5   1 110820          Y  253220 9153156 0.02766
## 6   1 111280          Y 2841898 9153156 0.31050</code></pre>
<pre class="r"><code>merged_nies_lmissing &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_nies_missing.lmiss&#39;, header = T)

head(merged_nies_lmissing)</code></pre>
<pre><code>##   CHR         SNP N_MISS N_GENO  F_MISS
## 1   1  rs62637813     47    361 0.13020
## 2   1 rs146477069     75    361 0.20780
## 3   1 rs141149254    188    361 0.52080
## 4   1   rs2462492    213    361 0.59000
## 5   1 rs143174675     27    361 0.07479
## 6   1   rs3091274     16    361 0.04432</code></pre>
</div>
</div>
</div>
