---
title: Excluding low frequency variants and genomic PCA
author: Martha Aquino
date: '2018-11-07'
slug: excluding-low-frequency-variants-and-genomic-pca
categories:
  - Coding
  - Experiments
tags:
  - Data Cleaning
  - PCA
  - PLINK
---



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Despite including age, sex, and population structure as covariates in the heritability and GWAS models, the results still seemed to be inflated. This was indicated by the many variants passing genome-wide significance, particularly for Component 1, 3, and R Cyl. Prior to adding the covariates, the QQ plots also displayed shelving, indicating that there were more variants passing the threshold than expected. Although, this alone could be normal and indicative of many variants in LD, in conjunction with the messy peaks on the manhattan plot, the results seem suspicious.</p>
<p>Therefore, I will exclude low frequency variants (0.01 &lt; MAF &lt; 0.05) and re-run all genomic analysis starting with the PCA to assess population structure.</p>
</div>
<div id="methods" class="section level1">
<h1>Methods</h1>
<div id="remove-low-frequency-variants" class="section level4">
<h4>1. Remove low frequency variants</h4>
<pre class="bash"><code>plink1.9 --bfile merged_nies/merged_nies_geno_210818 --maf 0.05 --make-bed --out merged_nies/merged_nies_071118</code></pre>
<ul>
<li>merged_nies_geno_210818 was the final data set used for analysis. It had been merged and then filtered with HWE and missingness. Data set contained 345 individuals and 3,910,077 variants.</li>
</ul>
<p>*Removing low freq variants removed 1,389,891 variants</p>
<p>*2,520,186 variants and 345 individuals remain</p>
</div>
<div id="run-pca-on-new-data-set" class="section level4">
<h4>2. Run PCA on new data set</h4>
<pre class="bash"><code>plink1.9 --bfile merged_nies/merged_nies_geno071118 --pca 20 var-wts --out plink_output/nies_geno_pca071118</code></pre>
</div>
<div id="load-eigenvalue-file-and-generate-screeplot" class="section level4">
<h4>3. Load eigenvalue file and generate screeplot</h4>
<pre class="r"><code>nies_eigenval071118 &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/nies_geno_pca071118.eigenval&#39;, header = F)

barplot(nies_eigenval071118$V1, 
        names.arg = 1:nrow(nies_eigenval071118), 
        main = &quot;NIES PCA Eigenvalue&quot;,
        xlab = &quot;Principal Components&quot;,
        ylab = &quot;Eigenvalue&quot;,
        col =&quot;lightskyblue2&quot;)
lines(x = 1:nrow(nies_eigenval071118), nies_eigenval071118$V1, 
      type = &quot;b&quot;, pch = 19, col = &quot;red&quot;)</code></pre>
<p><img src="/posts/2018-11-07-excluding-low-frequency-variants-and-genomic-pca_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
</div>
<div id="load-eigenvector-file-and-generate-pca-plot" class="section level4">
<h4>4. Load eigenvector file and generate PCA plot</h4>
<pre class="r"><code>nies_eigenvec071118 &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/nies_geno_pca071118.eigenvec&#39;, header = F)

plot(nies_eigenvec071118$V3, nies_eigenvec071118$V4, xlab = &quot;PC1&quot;, ylab = &quot;PC2&quot;, main = &quot;PC1 vs PC2 eigenvectors&quot;)</code></pre>
<p><img src="/posts/2018-11-07-excluding-low-frequency-variants-and-genomic-pca_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre class="r"><code>plot(nies_eigenvec071118$V4, nies_eigenvec071118$V5, xlab = &quot;PC2&quot;, ylab = &quot;PC3&quot;, main = &quot;PC2 vs PC3 eigenvectors&quot;)</code></pre>
<p><img src="/posts/2018-11-07-excluding-low-frequency-variants-and-genomic-pca_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<pre class="r"><code>plot(nies_eigenvec071118$V5, nies_eigenvec071118$V6, xlab = &quot;PC3&quot;, ylab = &quot;PC4&quot;, main = &quot;PC3 vs PC4 eigenvectors&quot;)</code></pre>
<p><img src="/posts/2018-11-07-excluding-low-frequency-variants-and-genomic-pca_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>Interestingly, the clustering of individuals that was observed in the previous PCA (without filtering for MAF) remained consistent for the PC1 vs PC2 plot.</p>
<p>Removing low frequency and rare variants did not seem to have a significant effect on the PCA results. The plots are still suggestive of population stratification most likely due to the genetic admixture.</p>
</div>
</div>
