---
title: Generating basic statistics for genomic data sets
author: Martha Aquino
date: '2018-06-27'
slug: generating-basic-statistics-for-genomic-data-sets
categories:
  - Coding
  - Experiments
tags:
  - Genomic Data
  - Linux
  - PLINK
---



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>In preparation for merging the two sets of genomic data, some basic statistics must first be generated. These will include allele frequencies, genotyping rates, missingness, and hardy-weinberg equilibrium tests.</p>
</div>
<div id="methods" class="section level1">
<h1>Methods</h1>
<div id="generate-allele-frequencies-for-both-files" class="section level2">
<h2>1.Generate allele frequencies for both files</h2>
<pre class="bash"><code>
plink1.9 --bfile NI_wgs_merged_snps --freq --out plink_output/wgs_allele_freq

plink1.9 --bfile NImerged.impute2.chrAllX-2014-Oct-02 --freq --out plink_output/snp_allele_freq
</code></pre>
<div id="view-head-of-output" class="section level3">
<h3>View head of output</h3>
</div>
</div>
<div id="plot-histograms-of-allele-frequency-distributions-in-both-files" class="section level2">
<h2>2.Plot histograms of allele frequency distributions in both files</h2>
</div>
<div id="generate-missingness-statistics" class="section level2">
<h2>3.Generate missingness statistics</h2>
<pre class="bash"><code>
plink1.9 --bfile NI_wgs_merged_snps --missing --out plink_output/wgs_missing

plink1.9 --bfile NImerged.impute2.chrAllX-2014-Oct-02 --missing --out plink_output/snp_missing</code></pre>
</div>
<div id="generate-hardy-weinberg-test-statistics" class="section level2">
<h2>4.Generate hardy-weinberg test statistics</h2>
<pre class="bash"><code>plink1.9 --bfile NI_wgs_merged_snps --missing --out plink_output/wgs_hardy

plink1.9 --bfile NImerged.impute2.chrAllX-2014-Oct-02 --hardy --out plink_output/snp_hardy</code></pre>
</div>
<div id="filter-out-variants-based-on-hwe-p-value" class="section level2">
<h2>5.Filter out variants based on HWE p-value</h2>
<pre class="bash"><code>plink1.9 --bfile NI_wgs_merged_snps --hwe 1.8e-7 --make-bed --out plink_output/wgs_hwe_filter

plink1.9 --bfile NImerged.impute2.chrAllX-2014-Oct-02 --hwe 1.8e-7 --make-bed --out plink_output/snp_hwe_filter</code></pre>
<div id="note-new-bed-bim-fam-files-created." class="section level3">
<h3>Note: new bed, bim, fam files created.</h3>
</div>
</div>
<div id="try-merging-to-find-variants-with-3-alleles" class="section level2">
<h2>6.Try merging to find variants with 3+ alleles</h2>
<pre class="bash"><code>plink1.9 --bfile plink_output/wgs_hwe_filter --bmerge plink_output/snp_hwe_filter.bed plink_output/snp_hwe_filter.bim plink_output/snp_hwe_filter.fam --make-bed --out plink_output/trial_merge</code></pre>
<div id="failed-merge-generated-a-file-that-has-a-list-of-variant-ids-that-have-3-alleles.-this-can-be-used-to-exlude-these-variants-from-each-file-and-the-merge-can-be-attempted-again.-note-6528-variants-were-found-to-have-multiple-alleles." class="section level3">
<h3>Failed merge generated a file that has a list of variant IDs that have 3+ alleles. This can be used to exlude these variants from each file and the merge can be attempted again. Note: 6528 variants were found to have multiple alleles.</h3>
</div>
</div>
<div id="exclude-variants-with-3-alleles-from-both-files" class="section level2">
<h2>7.Exclude variants with 3+ alleles from both files</h2>
<pre class="bash"><code>plink1.9 --bfile plink_output/wgs_hwe_filter --exclude plink_output/trial_merge-merge.missnp --make-bed --out plink_output/wgs_exclude_out

plink1.9 --bfile plink_output/snp_hwe_filter --exclude plink_output/trial_merge-merge.missnp --make-bed --out plink_output/snp_exclude_out
</code></pre>
<div id="output-from-exclude-wgs-9288730-variants-remain-from-9295158-snp-27303763-variants-remain-from-27310391.-note-new-bed-bim-fam-files-created." class="section level3">
<h3>Output from exclude: WGS = 9,288,730 variants remain (from 9,295,158); SNP = 27,303,763 variants remain (from 27,310,391). Note: new bed, bim, fam files created.</h3>
</div>
</div>
<div id="try-merge-again-with-new-binary-files-multi-allelic-variants-excluded" class="section level2">
<h2>8.Try merge again with new binary files (multi-allelic variants excluded)</h2>
<pre class="bash"><code>plink1.9 --bfile plink_output/wgs_exclude_out --bmerge plink_output/snp_exclude_out.bed plink_output/snp_exclude_out.bim plink_output/snp_exclude_out.fam --make-bed --out plink_output/var_out_merge</code></pre>
<div id="input-9288730-variants-loaded-from-wgs_exclude_out-27303763-variants-loaded-from-snp_excluded_out" class="section level3">
<h3>Input: 9,288,730 variants loaded from WGS_exclude_out; 27,303,763 variants loaded from SNP_excluded_out</h3>
</div>
<div id="variants-are-common-in-both-21516064-variants-are-new" class="section level3">
<h3>5,787,699 variants are common in both, 21,516,064 variants are new</h3>
</div>
<div id="warnings-5731755-variants-with-multiple-positions-69831-variants-with-same-positions" class="section level3">
<h3>Warnings: 5,731,755 variants with multiple-positions; 69,831 variants with same-positions</h3>
</div>
<div id="output-total-genotyping-rate-is-77.35-30804794-variants-and-614-individuals-pass-filters-and-qc" class="section level3">
<h3>Output: Total genotyping rate is 77.35%; 30,804,794 variants and 614 individuals pass filters and QC</h3>
</div>
</div>
</div>
