---
title: MLMA GWAS for most heritable trait
author: Martha Aquino
date: '2018-09-06'
slug: mlma-gwas-for-most-heritable-trait
categories:
  - Coding
  - Experiments
tags:
  - GCTA
  - Genomic Data
  - GWAS
  - MLMA
  - GREML
  - Linux
---



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>After obtaining heritability estimates for all traits and principal components, I can prioritise the traits for GWAS. There are different methods for performing GWAS and analysing GWAS results in GCTA. In particular, I am interested in the MLMA method which is mixed linear model associations (<a href="https://www.ncbi.nlm.nih.gov/pubmed/24473328" class="uri">https://www.ncbi.nlm.nih.gov/pubmed/24473328</a>).</p>
</div>
<div id="methods-and-results" class="section level1">
<h1>Methods and Results</h1>
<div id="run-gwas-mlma-on-r-k-value-v" class="section level3">
<h3>1. Run GWAS-MLMA on R K-value V</h3>
<pre class="bash"><code>./gcta64 --mlma --bfile merged_nies/merged_nies_geno_210818 --grm gcta_output/merged_nies_grm --pheno nies_pheno_020918.txt --mpheno 11 --out gcta_output/r_kval_mlma</code></pre>
<pre class="r"><code>r_kvalV_mlma &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/gcta_output/r_kval_mlma.mlma&#39;, header = TRUE)

head(r_kvalV_mlma)</code></pre>
<pre><code>##   Chr         SNP     bp A1 A2       Freq           b       se        p
## 1   1   rs2808353 590318  G  A 0.01449280 -0.06731680 0.534089 0.899700
## 2   1 rs138660747 778897  A  C 0.00434783 -0.16304800 1.113150 0.883547
## 3   1  rs10751453 783175  T  C 0.00000000         NaN      Inf      NaN
## 4   1   rs3131973 804046  A  G 0.00000000         NaN      Inf      NaN
## 5   1   rs3094315 817186  G  A 0.16666700 -0.14247400 0.191881 0.457777
## 6   1  rs12562034 833068  A  G 0.13043500 -0.00872999 0.194708 0.964238</code></pre>
</div>
<div id="run-gwas-mlma-on-uvaf-for-comparison" class="section level3">
<h3>2.Run GWAS-MLMA on UVAF for comparison</h3>
<pre class="bash"><code>./gcta64 --mlma --bfile merged_nies/merged_nies_geno_210818 --grm gcta_output/merged_nies_grm --pheno nies_pheno_020918.txt --mpheno 27 --out gcta_output/uvaf_mlma</code></pre>
<pre class="r"><code>uvaf_mlma &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/gcta_output/uvaf_mlma.mlma&#39;, header = TRUE)

head(uvaf_mlma)</code></pre>
<pre><code>##   Chr         SNP     bp A1 A2       Freq        b       se        p
## 1   1   rs2808353 590318  G  A 0.01449280 8.332810  9.08738 0.359161
## 2   1 rs138660747 778897  A  C 0.00434783 5.228320 17.82250 0.769251
## 3   1  rs10751453 783175  T  C 0.00000000      NaN      Inf      NaN
## 4   1   rs3131973 804046  A  G 0.00000000      NaN      Inf      NaN
## 5   1   rs3094315 817186  G  A 0.16666700 0.753981  3.24965 0.816523
## 6   1  rs12562034 833068  A  G 0.13043500 1.428780  3.27064 0.662220</code></pre>
</div>
<div id="re-run-gwas-on-uvaf-with-grm-and-including-2pcs" class="section level3">
<h3>3. Re-run GWAS on UVAF with GRM and including 2PCs</h3>
<pre class="r"><code>uvaf_2pcs_mlma &lt;- read.table(&#39;C:/Users/Martha/Documents/Honours/Project/honours.project/Data/gcta_output/uvaf_2pcs_mlma.mlma&#39;, header = TRUE)

head(uvaf_2pcs_mlma)</code></pre>
<pre><code>##   Chr         SNP     bp A1 A2       Freq        b       se        p
## 1   1   rs2808353 590318  G  A 0.01449280 8.387490  9.12358 0.357928
## 2   1 rs138660747 778897  A  C 0.00434783 5.843380 17.96990 0.745048
## 3   1  rs10751453 783175  T  C 0.00000000      NaN      Inf      NaN
## 4   1   rs3131973 804046  A  G 0.00000000      NaN      Inf      NaN
## 5   1   rs3094315 817186  G  A 0.16666700 0.732135  3.26355 0.822495
## 6   1  rs12562034 833068  A  G 0.13043500 1.230080  3.28702 0.708238</code></pre>
</div>
<div id="generate-manhattan-plots-for-uvaf" class="section level3">
<h3>4. Generate manhattan plots for UVAF</h3>
<pre class="r"><code>require(qqman)</code></pre>
<pre><code>## Loading required package: qqman</code></pre>
<pre><code>## </code></pre>
<pre><code>## For example usage please run: vignette(&#39;qqman&#39;)</code></pre>
<pre><code>## </code></pre>
<pre><code>## Citation appreciated but not required:</code></pre>
<pre><code>## Turner, S.D. qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots. biorXiv DOI: 10.1101/005165 (2014).</code></pre>
<pre><code>## </code></pre>
<pre class="r"><code>require(magrittr)</code></pre>
<pre><code>## Loading required package: magrittr</code></pre>
<pre class="r"><code>require(dplyr)</code></pre>
<pre><code>## Loading required package: dplyr</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>manhattan(uvaf_mlma, chr=&quot;Chr&quot;, bp=&quot;bp&quot;, p=&quot;p&quot;, snp=&quot;SNP&quot;, main = &quot;UVAF&quot;, ylim = c(0,10))</code></pre>
<div class="figure">
<img src="C:/Users/Martha/Documents/Honours/Project/electronic_lab_notebook-master/images/uvaf_notfiltered.jpeg" alt="UVAF not filtered" />
<p class="caption">UVAF not filtered</p>
</div>
<p>Evidently from this plot, there appears to be a prominent line of variants above -log(p) = 8. This is highly unusual and appears even more prominently when R K-value V is plotted (see below; data generated by Miles in gaston)</p>
<div class="figure">
<img src="C:/Users/Martha/Documents/Honours/Project/electronic_lab_notebook-master/images/r_kvalV-gaston.png" alt="R K-value V not filtered" />
<p class="caption">R K-value V not filtered</p>
</div>
<p>Rod suspected that the lines of variants may be due to rare variants. Rare variants are usually not included in a conventional GWAS/pedigree-based GWAS.</p>
</div>
<div id="filter-out-rare-variants-maf-0.01-and-variants-with--log10p-1-and-re-generate-manhattan-plots" class="section level3">
<h3>5. Filter out rare variants (MAF &lt; 0.01) and variants with -log10(p) &lt; 1 and re-generate manhattan plots</h3>
<pre class="r"><code>uvaf_mlma2 &lt;- uvaf_mlma %&gt;% filter(-log10(p)&gt;1) %&gt;% filter(Freq &gt; 0.01)
manhattan(uvaf_mlma2, chr=&quot;Chr&quot;, bp=&quot;bp&quot;, p=&quot;p&quot;, snp=&quot;SNP&quot;, main = &quot;UVAF&quot;, ylim = c(0,10))</code></pre>
<p><img src="/posts/2018-09-05-mlma-vs-greml-gwas-for-most-heritable-trait_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<pre class="r"><code>r_kvalV_mlma2 &lt;- r_kvalV_mlma %&gt;% filter(-log10(p)&gt;1) %&gt;% filter(Freq &gt; 0.01)
manhattan(r_kvalV_mlma2, chr=&quot;Chr&quot;, bp=&quot;bp&quot;, p=&quot;p&quot;, snp=&quot;SNP&quot;, main = &quot;R K-value V&quot;, ylim = c(0,10))</code></pre>
<p><img src="/posts/2018-09-05-mlma-vs-greml-gwas-for-most-heritable-trait_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>Filtering out the variants with MAF &lt; 0.01 removes the lines of variants and produces manhattan plots that are more acceptable.</p>
</div>
</div>
