---
title: Merging genomic data sets
author: Martha Aquino
date: '2018-07-25'
slug: merging-genomic-data-sets
categories:
  - Coding
  - Experiments
tags:
  - Data Cleaning
  - Genomic Data
  - Linux
  - PLINK
---

#Introduction
Merging the two genomic data sets caused an error because my laptop has inadequate RAM to perform the merge in a single command. Therefore, I will split the data and merge them by chromosome and stitch these together to create a unified merged file set. 

#Method

###1. Use 'for' loop to split SNP data by chromosome
```{bash, eval = FALSE}
for i in {1..23}
  do plink1.9 --bfile plink_output/snp_exclude --chr $i --make-bed --out merged_chr_data/snp_chr$i
done
```

###2. Repeat 'for' loop to split WGS data by chromosome
```{bash, eval = FALSE}
for i in {1..23}
  do plink1.9 --bfile plink_output/wgs_exclude --chr $i --make-bed --out merged_chr_data/wgs_chr$i
done
```

###3. Merge SNP and WGS data by chromosome 
```{bash, eval = FALSE}
for i in {1..23}
  do plink1.9 --bfile merged_chr_data/wgs_chr$i --bmerge merged_chr_data/snp_chr$i.bed merged_chr_data/snp_chr$i.bim merged_chr_data/snp_chr$i.fam --make-bed --out merged_chr_data/merged_chr$i 
done
```

###5. Create .txt file with the file names of all the merged file sets
For example: 
merged_chr_data/merged_chr2.bed merged_chr_data/merged_chr2.bim  merged_chr_data/merged_chr2.fam
merged_chr_data/merged_chr3.bed merged_chr_data/merged_chr3.bim  merged_chr_data/merged_chr3.fam


###4. Stitch together all merged files (chr 1 -23)
```{bash, eval = FALSE}
plink1.9 --bfile merged_chr_data/merged_chr1 --merge-list merged_chr_data/all_merged.txt --make-bed --out plink_output/var_out_merge
```

This produced the same 'out of memory' error that I received when I attempted to merge the original file sets. To try and resolve the issue, I increased the RAM for PLINK workspace from 8GB to 12GB by: 

```{bash, eval = FALSE}
plink1.9 --bfile merged_chr_data/merged_chr1 --merge-list merged_chr_data/all_merged_chr.txt --memory 12000 --make-bed --out plink_output/var_out_merge
```

I did not let this run finish as it was producing numerous 'multiple chromosomes' warnings. Miles has realised that the SNP array data was imputed using the hg19 build whereas the WGS was with hg38. SNPs can change (move or disappear) between builds, thus the 'multiple locations/chromosomes' warning. 

#Results 

I took note of the number of variants and genotyping rates that resulted from splitting and merging the data sets. 

```{r}
chr_var_merge <- read.csv('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/variant_chr_split.csv', header = T)

chr_var_merge
```

I will abandon this exercise because the SNP array data has to be changed to the hg38 build. 