---
title: Re-trying default merge
author: Martha Aqquino
date: '2018-08-14'
slug: re-trying-default-merge
categories:
  - Coding
  - Experiments
tags:
  - Data Cleaning
  - Genomic Data
  - PLINK
  - R
---



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The missing genotype filtering is still removing more variants than expected. We suspect that PLINK may not be handling the overlapping data/duplicate samples well. Therefore, I will be removing the SNP-array data for individuals that have duplicate data before attempting to merge the data set using the default setting (consensus calls) in PLINK.</p>
</div>
<div id="methods-and-results" class="section level1">
<h1>Methods and Results</h1>
<div id="identify-individuals-with-snp-array-only-and-duplicate-data." class="section level3">
<h3>1. Identify individuals with SNP-array only and duplicate data.</h3>
<p>Using the master spreadsheet containing UUIDs, I 1) isolated all NIES individuals, 2) isolated NIES individuals with SNP data (GWAS_NIES), 3) separate individuals with potential duplicates and SNP-array data.</p>
<p>From this:</p>
<ul>
<li><p>363 UUIDs with SNP-data</p></li>
<li><p>73 UUIDs with WGS and/or SNP-array (I do not know for sure which individuals have duplicate data because I do not have the list of IDs for individuals with SNP-array data)</p></li>
<li><p>290 UUIDs with ONLY SNP-array data (presumably)</p></li>
</ul>
</div>
<div id="extract-snp-aray-data-for-290-individuals" class="section level3">
<h3>2. Extract SNP-aray data for 290 individuals</h3>
<p>array_filtered file contains 9million+ variants extracted from the original SNP-array file.</p>
<pre class="bash"><code>plink --bfile plink_output/array_filtered --keep snp-array_only_uuid.txt --make-bed --out NIES.array.hg38</code></pre>
<p>Note: data for 288 are extracted</p>
</div>
<div id="try-default-merging-with-new-array-data-set" class="section level3">
<h3>3. Try default merging with new array data set</h3>
<pre class="bash"><code>plink --bfile NIES.array.hg38 --bmerge plink_output/wgs_filtered.bed plink_output/wgs_filtered.bim plink_output/wgs_filtered.fam --make-bed --out merged_def2_nies</code></pre>
<p>I accidentally used the wgs_filtered file which contains data for all 108 individuals with WGS data. Not all are NIES individuals.</p>
</div>
<div id="extract-merged-data-for-nies" class="section level3">
<h3>4. Extract merged data for NIES</h3>
<pre class="bash"><code>plink --bfile merged_def2_nies --keep gwas_niesID.txt --make-bed --out merged_def3_nies</code></pre>
<p>gwas_niesID.txt contains UUIDs of all NIES individuals with SNP data.</p>
<p>Output:</p>
<ul>
<li>361 people remain</li>
<li>9,155,053 variants</li>
</ul>
</div>
<div id="fix-paternal-and-maternal-ids-in-merged_def3_nies-fam-file-change-all-to-0" class="section level3">
<h3>5. Fix paternal and maternal IDs in merged_def3_nies fam file (change all to 0)</h3>
</div>
<div id="filter-variants-based-on-hwe-p-value" class="section level3">
<h3>6. Filter variants based on HWE p-value</h3>
<pre class="bash"><code>plink1.9 --bfile merged_def3_nies --hwe 1.8e-7 --make-bed --out plink_output/merged_nies_hwefilter</code></pre>
<p>Output:</p>
<ul>
<li>8811 variants removed</li>
<li>9,146,242 variants remain</li>
</ul>
</div>
<div id="filter-based-on-missing-genotypes" class="section level3">
<h3>7. Filter based on missing genotypes</h3>
<pre class="bash"><code>plink1.9 --bfile plink_output/merged_nies_hwefilter --mind 0.05 --geno 0.01 --make-bed --out merged_nies_filter</code></pre>
<p>Output:</p>
<ul>
<li>2 people removed</li>
<li>3,158,627 variants removed</li>
</ul>
<pre class="bash"><code>plink1.9 --bfile plink_output/merged_nies_hwefilter --mind 0.1 --geno 0.05 --make-bed --out merged_nies_filter
</code></pre>
<p>Output:</p>
<ul>
<li>0 people removed</li>
<li>1,092,421 variants removed</li>
</ul>
<p>Trying the default merge yielded very similar results to merge mode 3 with overlapping results.</p>
</div>
</div>
