---
title: Generating stats for merged data and extracting data for NIES
author: Martha Aquino
date: '2018-08-03'
slug: generating-stats-for-merged-data-and-extracting-data-for-nies
categories:
  - Coding
  - Experiments
tags:
  - Data Cleaning
  - Genomic Data
  - Linux
  - PLINK
---

#Introduction
Now that the merge has been successful and there should no longer be duplicated individuals, general stats need to be performed on the merged data set. Subsequently, the final step to this data prep and cleaning is to extract data for the NIES individuals only. 

##Methods and Results 
Note: Most processes up until the merge from the previous entry was performed on taurus. I transferred filtered_merge to local directory. These stat tests will be performed locally. 
###1. Generate allele freq stats
```{bash, eval = FALSE}
plink --bfile plink_output/filtered_merge --freq --out plink_output/merge_allele_freq
```

###2. Load allele freq file
```{r}
merged_allele_freq <- read.table('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_allele_freq.frq', header = T)

head(merged_allele_freq)
```

###3. Generate histogram for allele freq distributions
```{r}
hist(merged_allele_freq$MAF, main = "Allele Frequency Distributions of Merged Data", xlab = "Minor Allele Frequency", xlim = c(0.0, 0.5), col = "plum2")
```

###4. Generate missingness statistics
```{bash, eval = FALSE}
plink1.9 --bfile plink_output/filtered_merge --missing --out plink_output/merged_missing
```
```{r}
merged_missing_snp <- read.table('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_missing.lmiss', header = T)

head(merged_missing_snp)

merged_missing_indiv <- read.table('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_missing.imiss', header = T)

head(merged_missing_indiv)
```


###5. Generate hardy-weinberg statistics
```{bash, eval = FALSE}
plink1.9 --bfile plink_output/filtered_merge --hardy --out plink_output/merged_hardy
```
```{r}
#merged_hardy <- read.table('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_hardy.hwe', header = T)

#head(merged_hardy)
```


###6. Filter out variants based on HWE p-value
```{bash, eval = FALSE}
plink1.9 --bfile plink_output/filtered_merge --hwe 1.8e-7 --make-bed --out plink_output/merged_hwe_filter
```
Note: unlike previous attempts at generating stats before merging, I did not use a HWE p-val filter before merging this data set. 
Note: this creates a new file set

Output:
  - 1897 variants were excluded based on hwe p-val
  - 9,153,156 variants remain
  - warning: hwe observation counts vary by more than 10%
  
###7. Check allele freq stats after filtering
```{bash, eval = FALSE}
plink1.9 --bfile plink_output/merged_hwe_filter --freq --out plink_output/merge_hwefilter_freq
```

###8. Load allele freq file
```{r}
merged_hwefilter_freq <- read.table('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merge_hwefilter_freq.frq', header = T)
```

###9. Generate histogram of allele freq distributions
```{r}
hist(merged_hwefilter_freq$MAF, main = "Allele Frequency Distributions of Merged Data after HWE filtering", xlab = "Minor Allele Frequency", xlim = c(0.0, 0.5), col = "peachpuff")
```

###10. Load spreadsheet with NI study participants
```{r}
ni_ped_uuid <- read.csv('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/NI_UUID_Ped_20180718.csv', header = T) 

head(ni_ped_uuid)
```

###11. Isolate all NIES individuals using "NIES" column, and all NIES indivs with sequencing data using "GWAS_NIES" column 
```{r}
gwas_niesID <- read.csv('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/gwas_niesID.csv', header = F)

head(gwas_niesID)
```
There are 363 NIES individuals that may have sequencing data identified from the spreadsheet. This list was also converted to .txt to be compatible with PLINK.

###12. Extract data in PLINK
```{bash, eval = FALSE}
plink1.9 --bfile plink_output/merged_hwe_filter --keep gwas_niesID.txt --make-bed --out merged_nies
```
This extracted data for only 361 individuals, which is 2 less than expected. I suspect it may be because of differing paternal and maternal IDs. Therefore, I changed all maternal and paternal IDs in "merged_hwe_filter.fam" to 0s but this did not resolve the issue. PLINK is still only extracting data for 361 individuals (9,153,156 variants; total genotyping rate = 91.68%)

The file sets created in this step will be used for all subsequent analysis. 

###13. Check allele frequencies of this final data set
```{bash, eval = FALSE}
plink1.9 --bfile merged_nies --freq --out plink_output/merged_nies_allele_freq
```

###14. Load allele freq file
```{r}
merged_nies_allele_freq <- read.table('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_nies_allele_freq.frq', header = T)

head(merged_nies_allele_freq)
```

###15. Generate histogram for allele frequency distributions
```{r}
hist(merged_nies_allele_freq$MAF, main = "Allele Frequency Distributions of NIES Data", xlab = "Minor Allele Frequency", xlim = c(0.0, 0.5), col = "lightskyblue1")
```

###16. Check missingness rates
```{bash, eval = FALSE}
plink1.9 --bfile merged_nies --missing --out plink_output/merged_nies_missing
```
```{r}
merged_nies_imissing <- read.table('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_nies_missing.imiss', header = T)

head(merged_nies_imissing)

merged_nies_lmissing <- read.table('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/merged_nies_missing.lmiss', header = T)

head(merged_nies_lmissing)
```

###17. The current data set should be filtered for missingness rates. 
PLINK's default parameters excludes individuals with >10% missing genotypes, and SNPs with >10% missing calls. According to literature, typical QC excludes individuals with >2-7% missing genotypes, and SNPs with >1-5% mising calls. (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3061487/) (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3025522/)
```{bash, eval = FALSE}
plink1.9 --bfile merged_nies --mind 0.1 --geno 0.05 --make-bed --out merged_nies_miss_filter
```

This filters out individuals with >10% missing genotypes and SNPs with >5% missing call rates. By doing so, 73 individuals and 1,357,838 variants were filtered out. The new data set contains: 288 individuals and 7,795,318 variants, with a total genotpying rate of 97.10% (this genotyping rate has increased from 91.68%).

This filtering step excluded more individuals and variants than I expected. The exlcuded individuals had missing genotyping rates ranging from 19 - 31%. However, due to the modest number of samples in this study, we are reluctant to exclude any individuals and will therefore not be using a filter for missing genotypes per sample.

###18. Apply genotyping rate filter for SNPs.
```{bash, eval = FALSE}
plink1.9 --bfile merged_nies --geno 0.1 --make-bed --out merged_nies_miss_filter
```

This filters out 3,465,839 variants. The final data set includes 361 individuals and 5,687,317 variants with a total genotyping rate of 91.68%. 

###19. Check allele frequencies for this filtered data set
```{bash, eval = FALSE}
plink1.9 --bfile merged_nies_miss_filter --freq --out plink_output/nies_geno0.1_allele_freq
```

Output:
   - 361 individuals
   - 5,687,317 variants
   - total genotyping rate = 95.78%
   
###20. Load allele frequency file 
```{r}
nies_geno0.1_freq <- read.table('C:/Users/Martha/Documents/Honours/Project/honours.project/Data/plink_output/nies_geno0.1_allele_freq.frq', header = T)

head(nies_geno0.1_freq)
```

###21. Generate histogram for allele frequency distribution
```{r}
hist(nies_geno0.1_freq$MAF, main = "Allele Frequency Distributions of NIES Data Post-Geno Filtering", xlab = "Minor Allele Frequency", xlim = c(0.0, 0.5), col = "salmon1")

```

The allele frequency distribution of this filtered data set appears to retain roughly the same number of rare variants (MAR < 0.05). However, there is a noticeable change 